<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Data Manipulation and Storage | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.10 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Data Manipulation and Storage | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Data Manipulation and Storage | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Jingwen Yang" />


<meta name="date" content="2019-06-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html">
<link rel="next" href="pairwise-expression-distance-and-expression-tree-making.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"><i>TreeExp</i>-Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#scope"><i class="fa fa-check"></i><b>2.1</b> Scope</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#citation"><i class="fa fa-check"></i><b>2.3</b> Citation</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#how-to-get-help"><i class="fa fa-check"></i><b>2.4</b> How to Get Help?</a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#rna-seq-data-enbeded"><i class="fa fa-check"></i><b>2.5</b> RNA-seq Data Enbeded</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html"><i class="fa fa-check"></i><b>3</b> Data Manipulation and Storage</a><ul>
<li class="chapter" data-level="3.1" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#input-format"><i class="fa fa-check"></i><b>3.1</b> Input Format</a></li>
<li class="chapter" data-level="3.2" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#example-of-input-data"><i class="fa fa-check"></i><b>3.2</b> Example of Input Data</a></li>
<li class="chapter" data-level="3.3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#construction"><i class="fa fa-check"></i><b>3.3</b> Construction</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html"><i class="fa fa-check"></i><b>4</b> Pairwise Expression Distance and Expression Tree-Making</a><ul>
<li class="chapter" data-level="4.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#theory"><i class="fa fa-check"></i><b>4.1</b> Theory</a><ul>
<li class="chapter" data-level="4.1.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-ornstein-uhlenbeck-ou-model-of-transcriptome-evolution"><i class="fa fa-check"></i><b>4.1.1</b> The Ornstein-Uhlenbeck (OU) model of transcriptome evolution</a></li>
<li class="chapter" data-level="4.1.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#transcriptome-evolution-between-species"><i class="fa fa-check"></i><b>4.1.2</b> Transcriptome evolution between species</a></li>
<li class="chapter" data-level="4.1.3" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-estimation-problem-of-expression-distance"><i class="fa fa-check"></i><b>4.1.3</b> The estimation problem of expression distance</a></li>
<li class="chapter" data-level="4.1.4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-expression-distance-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>4.1.4</b> Estimation of expression distance under the stationary OU model</a></li>
<li class="chapter" data-level="4.1.5" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-distance-estimation-when-expression-optima-vary-among-genes"><i class="fa fa-check"></i><b>4.1.5</b> Expression distance estimation when expression optima vary among genes</a></li>
<li class="chapter" data-level="4.1.6" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-pi"><i class="fa fa-check"></i><b>4.1.6</b> Estimation of <span class="math inline">\(\pi\)</span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#case-study-expression-tree-building"><i class="fa fa-check"></i><b>4.2</b> Case Study: Expression Tree Building</a><ul>
<li class="chapter" data-level="4.2.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#distance-matrix"><i class="fa fa-check"></i><b>4.2.1</b> Distance matrix</a></li>
<li class="chapter" data-level="4.2.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-character-tree"><i class="fa fa-check"></i><b>4.2.2</b> Expression character tree</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html"><i class="fa fa-check"></i><b>5</b> Relative Rate Test for Transcriptome Evolution</a><ul>
<li class="chapter" data-level="5.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#theory-1"><i class="fa fa-check"></i><b>5.1</b> Theory</a></li>
<li class="chapter" data-level="5.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#statistical-procedure-of-the-relative-rate-test"><i class="fa fa-check"></i><b>5.2</b> Statistical Procedure of the Relative Rate Test</a><ul>
<li class="chapter" data-level="5.2.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#sampling-variance-of-the-expression-distance"><i class="fa fa-check"></i><b>5.2.1</b> Sampling variance of the expression distance</a></li>
<li class="chapter" data-level="5.2.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#calculation-of-varbigtriangleup_ab"><i class="fa fa-check"></i><b>5.2.2</b> Calculation of <span class="math inline">\(Var(\bigtriangleup_{AB})\)</span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#case-study-testing-fast-evolving-genes-in-human"><i class="fa fa-check"></i><b>5.3</b> Case Study: Testing Fast-evolving Genes in Human</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html"><i class="fa fa-check"></i><b>6</b> Estimating the Strength of Expression Conservation</a><ul>
<li class="chapter" data-level="6.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#theory-2"><i class="fa fa-check"></i><b>6.1</b> Theory</a><ul>
<li class="chapter" data-level="6.1.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stabilizing-selection-model-in-transcriptome-evolution-across-species"><i class="fa fa-check"></i><b>6.1.1</b> Stabilizing selection model in transcriptome evolution across species</a></li>
<li class="chapter" data-level="6.1.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stationary-ou-model-under-a-species-phylogeny"><i class="fa fa-check"></i><b>6.1.2</b> Stationary OU model under a species phylogeny</a></li>
<li class="chapter" data-level="6.1.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#variation-of-w-among-genes"><i class="fa fa-check"></i><b>6.1.3</b> Variation of <span class="math inline">\(W\)</span> among genes</a></li>
<li class="chapter" data-level="6.1.4" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#an-empirical-bayesian-framework-for-gene-specific-w-esitimation"><i class="fa fa-check"></i><b>6.1.4</b> An empirical Bayesian framework for gene-specific <span class="math inline">\(W\)</span> esitimation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#statistical-procedure"><i class="fa fa-check"></i><b>6.2</b> Statistical Procedure</a></li>
<li class="chapter" data-level="6.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#case-study-emprical-bayesian-esitimates-of-expression-conservation-of-genes"><i class="fa fa-check"></i><b>6.3</b> Case Study: Emprical Bayesian Esitimates of Expression Conservation of genes</a><ul>
<li class="chapter" data-level="6.3.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#inversed-correlation-matrix"><i class="fa fa-check"></i><b>6.3.1</b> Inversed correlation matrix</a></li>
<li class="chapter" data-level="6.3.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#estimation-of-gamma-parameters"><i class="fa fa-check"></i><b>6.3.2</b> Estimation of gamma parameters</a></li>
<li class="chapter" data-level="6.3.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#bayesian-estimation-of-gene-specific-selection-pressure"><i class="fa fa-check"></i><b>6.3.3</b> Bayesian estimation of gene-specific selection pressure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html"><i class="fa fa-check"></i><b>7</b> Ancestral Transcriptome Inference</a><ul>
<li class="chapter" data-level="7.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#theory-3"><i class="fa fa-check"></i><b>7.1</b> Theory</a><ul>
<li class="chapter" data-level="7.1.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#from-brownian-motion-bm-model-to-ornstein-uhlenbeck-ou-model"><i class="fa fa-check"></i><b>7.1.1</b> From Brownian Motion (BM) model to Ornstein-Uhlenbeck (OU) model</a></li>
<li class="chapter" data-level="7.1.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-ou-model-under-a-phylogeny"><i class="fa fa-check"></i><b>7.1.2</b> The OU model under a phylogeny</a></li>
<li class="chapter" data-level="7.1.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-gene-expression-inference-an-empirical-bayesian-approach"><i class="fa fa-check"></i><b>7.1.3</b> Ancestral gene expression inference: an empirical Bayesian approach</a></li>
<li class="chapter" data-level="7.1.4" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-problem-of-ancestral-transcriptome-inference"><i class="fa fa-check"></i><b>7.1.4</b> The problem of ancestral transcriptome inference</a></li>
<li class="chapter" data-level="7.1.5" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#stationary-ornstein-uhlenbeck-sou-model"><i class="fa fa-check"></i><b>7.1.5</b> Stationary Ornstein-Uhlenbeck (sOU) model</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#a-fast-algorithm-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>7.2</b> A Fast Algorithm under the Stationary OU Model</a></li>
<li class="chapter" data-level="7.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#case-study-ancestral-expression-inference-of-gene-emp1"><i class="fa fa-check"></i><b>7.3</b> Case Study: Ancestral Expression Inference of Gene <em>EMP1</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#expression-character-tree-1"><i class="fa fa-check"></i><b>7.3.1</b> Expression character tree</a></li>
<li class="chapter" data-level="7.3.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#creating-variance-co-variance-matrix"><i class="fa fa-check"></i><b>7.3.2</b> Creating variance co-variance matrix</a></li>
<li class="chapter" data-level="7.3.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-expression-estimation"><i class="fa fa-check"></i><b>7.3.3</b> Ancestral expression estimation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial: Use <em>TreeExp</em> for Phylogenetic Transcriptome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-manipulation-and-storage" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Data Manipulation and Storage</h1>
<div id="input-format" class="section level2">
<h2><span class="header-section-number">3.1</span> Input Format</h2>
<p>To simplify the use of package, the updated <em>TreeExp</em> has gave up the former input format (which require both gene information file and reads count data) and takes in only normalized RNA-seq data as input file. In other words, users should make sure that the input data is processed and comparable between samples. The package is not likely to provied data filteration or normalization functions.</p>
<p>The expression data is suppossed to be in certain format:</p>
<ol style="list-style-type: decimal">
<li><p>expression file should be a <code>text</code> file in matrix shape, in which values are separated by tabs. <code>Rows</code> correspond to orthologous genes names, and <code>Columns</code> correspond to sample names. Sample names are in format of “<code>TaxaName_SubtaxaName_ReplicatesName</code>”. Usually, <code>TaxaName</code> represents name of species; <code>Subtaxaname</code> correspond to cetain tissue, celltype or develomental stage; <code>ReplicatesName</code> shows the name of replicates for each <code>Taxa_Subtaxa</code>-pair. The three types of lables, <code>TaxaName</code>,<code>SubtaxaName</code>, <code>ReplicatesName</code> are connected by ‘<code>_</code>’ character.</p></li>
<li><p>raw reads count data should be first normalized, e.g., by RPKM. While RPKM is simple and straightforward, it tends to be unstable when the number of genes expressed across samples differs considerably. This problem can be alleviated by the TPM measure, which has been widely usued. Some statistically sophisticated normalization methods such as TMM, and median ratio normalization, has become the built-in standard in many bioinformatics tools for RNA-seq analysis<span class="citation">(Robinson, McCarthy, and Smyth <a href="#ref-robinson2010">2010</a>)</span>.</p></li>
</ol>
</div>
<div id="example-of-input-data" class="section level2">
<h2><span class="header-section-number">3.2</span> Example of Input Data</h2>
<p>The example file are included in the <em>TreeExp</em> package, which can be found in <code>/inst/extdata</code> folder in the package.</p>
<p>Here, we select expression values of 100 orthologous genes in eight brain regions (CB, HIP, STR, ACC, V1C, PMC, DPFC, VPFC)<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> among human, chimpanzee, gorilla and gibbon<span class="citation">(Xu et al. <a href="#ref-xu2018">2018</a>)</span>.
The numbers of biological replicates for each of the brain regions in species are 2~6, except only one replicate for all brain regions in gibbon.
Note that expression data here are only used as demonstration of how functions in package store, manipulate and print the data input, and should not be used in further phylogenetic analysis since too few genes were included in the file.</p>
<p>The Table below shows the format of the partly input data.</p>
<table>
<thead>
<tr class="header">
<th align="left">Gene</th>
<th align="right">Human_DPFC_Hs3</th>
<th align="right">Human_STR_Hs8</th>
<th align="right">Chimpanzee_ACC_REIKO</th>
<th align="right">Gorilla_CB_GON</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000000003</td>
<td align="right">10.5</td>
<td align="right">1.8</td>
<td align="right">3.7</td>
<td align="right">0.9</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000005</td>
<td align="right">0.0</td>
<td align="right">0.0</td>
<td align="right">0.0</td>
<td align="right">0.0</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000419</td>
<td align="right">33.7</td>
<td align="right">34.1</td>
<td align="right">17.3</td>
<td align="right">19.9</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000457</td>
<td align="right">1.3</td>
<td align="right">1.9</td>
<td align="right">1.2</td>
<td align="right">3.7</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000460</td>
<td align="right">0.5</td>
<td align="right">0.7</td>
<td align="right">0.6</td>
<td align="right">1.3</td>
</tr>
</tbody>
</table>
</div>
<div id="construction" class="section level2">
<h2><span class="header-section-number">3.3</span> Construction</h2>
<p>The construction function <code>TEconstruct</code> loads in expression level file, and wraps it in a list of <em>taxonExp</em> objects (one <em>taxaExp</em> object).</p>
<pre class="sourceCode r"><code class="sourceCode r">taxa.objects =<span class="st"> </span><span class="kw">TEconstruct</span>(<span class="dt">ExpValueFP =</span> <span class="kw">system.file</span>(<span class="st">&#39;extdata/primate_brain_expvalues.txt&#39;</span>,
<span class="dt">package =</span> <span class="st">&#39;TreeExp&#39;</span>), <span class="dt">taxa =</span> <span class="st">&quot;all&quot;</span>, <span class="dt">subtaxa =</span> <span class="st">&#39;all&#39;</span>)</code></pre>
<p>The construction process takes <strong>several minutes</strong> on a desktop computer depending on data size and hardware performance. Specify <strong>“taxa”</strong> and <strong>“subtaxa”</strong> options in the function when using partial of your data. The construction process will be faster.</p>
<pre class="sourceCode r"><code class="sourceCode r">taxa.objects =<span class="st"> </span><span class="kw">TEconstruct</span>(<span class="dt">ExpValueFP =</span> <span class="kw">system.file</span>(<span class="st">&#39;extdata/primate_brain_expvalues.txt&#39;</span>,
<span class="dt">package =</span> <span class="st">&#39;TreeExp&#39;</span>), <span class="dt">taxa =</span> <span class="st">&quot;all&quot;</span>, <span class="dt">subtaxa =</span> <span class="kw">c</span>(<span class="st">&quot;ACC&quot;</span>,<span class="st">&quot;CB&quot;</span>))</code></pre>
<p>You can take a look at what the loaded objects:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(taxa.objects, <span class="dt">details =</span> <span class="ot">TRUE</span>)</code></pre>
<pre><code>## 
##  8 taxonExp objects 
## 
## object 1 : Human      ACC 
## object 2 : Human      CB 
## object 3 : Chimpanzee     ACC 
## object 4 : Chimpanzee     CB 
## object 5 : Gorilla    ACC 
## object 6 : Gorilla    CB 
## object 7 : Gibbon     ACC 
## object 8 : Gibbon     CB</code></pre>
<p>Also, you can choose to print a single <em>taxonExp</em> object</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(taxa.objects[[<span class="dv">1</span>]], <span class="dt">printlen =</span> <span class="dv">6</span>)</code></pre>
<pre><code>## 
## One taxonExp object
## Taxon name:  Human 
## Subtaxon name:  ACC 
## Total gene number:  200 
## Total bio replicates number:  5 
## Bio replicates titles:
## [1] &quot;Hs1&quot; &quot;Hs6&quot; &quot;Hs8&quot; &quot;Hs2&quot; &quot;Hs5&quot;</code></pre>
<p>and choose to print single element (<em>exp_val</em>(expression values) element as example) for the <em>taxonExp</em> object</p>
<pre class="sourceCode r"><code class="sourceCode r">taxa.objects[[<span class="dv">6</span>]]<span class="op">$</span>exp_value[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,]</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Gorilla_CB_Sakura</th>
<th align="right">Gorilla_CB_GON</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ENSG00000000003</td>
<td align="right">2.6</td>
<td align="right">0.9</td>
</tr>
<tr class="even">
<td>ENSG00000000005</td>
<td align="right">0.0</td>
<td align="right">0.0</td>
</tr>
<tr class="odd">
<td>ENSG00000000419</td>
<td align="right">13.5</td>
<td align="right">19.9</td>
</tr>
<tr class="even">
<td>ENSG00000000457</td>
<td align="right">4.1</td>
<td align="right">3.7</td>
</tr>
<tr class="odd">
<td>ENSG00000000460</td>
<td align="right">1.3</td>
<td align="right">1.3</td>
</tr>
</tbody>
</table>
<p>Once the contruction courcs successfully completed, the following transcriptome phylogenetic analysis are ready to go.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-robinson2010">
<p>Robinson, Mark D., Davis J. McCarthy, and Gordon K. Smyth. 2010. “edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.” <em>Bioinformatics</em> 26:139–40. <a href="https://doi.org/10.1093/bioinformatics/btp616" class="uri">https://doi.org/10.1093/bioinformatics/btp616</a>.</p>
</div>
<div id="ref-xu2018">
<p>Xu, Chuan, Qian Li, Olga Efimova, Liu He, Shoji Tatsumoto, Vita Stepanova, Takao Oishi, et al. 2018. “Human-specific features of spatial gene expression and regulation in eight brain regions.” <em>Genome Research</em> 28:1097–1110. <a href="https://doi.org/10.1101/gr.231357.117" class="uri">https://doi.org/10.1101/gr.231357.117</a>.</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>cerebellum (CB), hippocampus (HIP), striatum (STR) , anterior cingulate cortex (ACC), primary visual cortex (V1C), premotor cortex (PMC), dorsolateral prefrontal cortex (DPFC),
ventrolateral prefrontal cortex (VPFC).<a href="data-manipulation-and-storage.html#fnref1" class="footnote-back">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pairwise-expression-distance-and-expression-tree-making.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-Data.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["TreeExp-Tutorial.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
