<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Estimating the Strength of Expression Conservation | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.10 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Estimating the Strength of Expression Conservation | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Estimating the Strength of Expression Conservation | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Jingwen Yang" />


<meta name="date" content="2019-09-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="relative-rate-test-for-transcriptome-evolution.html">
<link rel="next" href="ancestral-transcriptome-inference.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"><i>TreeExp</i>-Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#scope"><i class="fa fa-check"></i><b>2.1</b> Scope</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#citation"><i class="fa fa-check"></i><b>2.3</b> Citation</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#how-to-get-help"><i class="fa fa-check"></i><b>2.4</b> How to Get Help?</a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#rna-seq-data-enbeded"><i class="fa fa-check"></i><b>2.5</b> RNA-seq Data Enbeded</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html"><i class="fa fa-check"></i><b>3</b> Data Manipulation and Storage</a><ul>
<li class="chapter" data-level="3.1" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#input-format"><i class="fa fa-check"></i><b>3.1</b> Input Format</a></li>
<li class="chapter" data-level="3.2" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#example-of-input-data"><i class="fa fa-check"></i><b>3.2</b> Example of Input Data</a></li>
<li class="chapter" data-level="3.3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#construction"><i class="fa fa-check"></i><b>3.3</b> Construction</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html"><i class="fa fa-check"></i><b>4</b> Pairwise Expression Distance and Expression Tree-Making</a><ul>
<li class="chapter" data-level="4.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#theory"><i class="fa fa-check"></i><b>4.1</b> Theory</a><ul>
<li class="chapter" data-level="4.1.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-ornstein-uhlenbeck-ou-model-of-transcriptome-evolution"><i class="fa fa-check"></i><b>4.1.1</b> The Ornstein-Uhlenbeck (OU) model of transcriptome evolution</a></li>
<li class="chapter" data-level="4.1.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#transcriptome-evolution-between-species"><i class="fa fa-check"></i><b>4.1.2</b> Transcriptome evolution between species</a></li>
<li class="chapter" data-level="4.1.3" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-estimation-problem-of-expression-distance"><i class="fa fa-check"></i><b>4.1.3</b> The estimation problem of expression distance</a></li>
<li class="chapter" data-level="4.1.4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-expression-distance-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>4.1.4</b> Estimation of expression distance under the stationary OU model</a></li>
<li class="chapter" data-level="4.1.5" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-distance-estimation-when-expression-optima-vary-among-genes"><i class="fa fa-check"></i><b>4.1.5</b> Expression distance estimation when expression optima vary among genes</a></li>
<li class="chapter" data-level="4.1.6" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-pi"><i class="fa fa-check"></i><b>4.1.6</b> Estimation of <span class="math inline">\(\pi\)</span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#case-study-expression-tree-building"><i class="fa fa-check"></i><b>4.2</b> Case Study: Expression Tree Building</a><ul>
<li class="chapter" data-level="4.2.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#distance-matrix"><i class="fa fa-check"></i><b>4.2.1</b> Distance matrix</a></li>
<li class="chapter" data-level="4.2.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-character-tree"><i class="fa fa-check"></i><b>4.2.2</b> Expression character tree</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html"><i class="fa fa-check"></i><b>5</b> Relative Rate Test for Transcriptome Evolution</a><ul>
<li class="chapter" data-level="5.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#theory-1"><i class="fa fa-check"></i><b>5.1</b> Theory</a></li>
<li class="chapter" data-level="5.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#statistical-procedure-of-the-relative-rate-test"><i class="fa fa-check"></i><b>5.2</b> Statistical Procedure of the Relative Rate Test</a><ul>
<li class="chapter" data-level="5.2.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#sampling-variance-of-the-expression-distance"><i class="fa fa-check"></i><b>5.2.1</b> Sampling variance of the expression distance</a></li>
<li class="chapter" data-level="5.2.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#calculation-of-varbigtriangleup_ab"><i class="fa fa-check"></i><b>5.2.2</b> Calculation of <span class="math inline">\(Var(\bigtriangleup_{AB})\)</span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#case-study-testing-fast-evolving-genes-in-human"><i class="fa fa-check"></i><b>5.3</b> Case Study: Testing Fast-evolving Genes in Human</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html"><i class="fa fa-check"></i><b>6</b> Estimating the Strength of Expression Conservation</a><ul>
<li class="chapter" data-level="6.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#theory-2"><i class="fa fa-check"></i><b>6.1</b> Theory</a><ul>
<li class="chapter" data-level="6.1.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stabilizing-selection-model-in-transcriptome-evolution-across-species"><i class="fa fa-check"></i><b>6.1.1</b> Stabilizing selection model in transcriptome evolution across species</a></li>
<li class="chapter" data-level="6.1.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stationary-ou-model-under-a-species-phylogeny"><i class="fa fa-check"></i><b>6.1.2</b> Stationary OU model under a species phylogeny</a></li>
<li class="chapter" data-level="6.1.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#variation-of-w-among-genes"><i class="fa fa-check"></i><b>6.1.3</b> Variation of <span class="math inline">\(W\)</span> among genes</a></li>
<li class="chapter" data-level="6.1.4" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#an-empirical-bayesian-framework-for-gene-specific-w-esitimation"><i class="fa fa-check"></i><b>6.1.4</b> An empirical Bayesian framework for gene-specific <span class="math inline">\(W\)</span> esitimation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#statistical-procedure"><i class="fa fa-check"></i><b>6.2</b> Statistical Procedure</a></li>
<li class="chapter" data-level="6.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#case-study-emprical-bayesian-esitimates-of-expression-conservation-of-genes"><i class="fa fa-check"></i><b>6.3</b> Case Study: Emprical Bayesian Esitimates of Expression Conservation of genes</a><ul>
<li class="chapter" data-level="6.3.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#inversed-correlation-matrix"><i class="fa fa-check"></i><b>6.3.1</b> Inversed correlation matrix</a></li>
<li class="chapter" data-level="6.3.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#estimation-of-gamma-parameters"><i class="fa fa-check"></i><b>6.3.2</b> Estimation of gamma parameters</a></li>
<li class="chapter" data-level="6.3.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#bayesian-estimation-of-gene-specific-selection-pressure"><i class="fa fa-check"></i><b>6.3.3</b> Bayesian estimation of gene-specific selection pressure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html"><i class="fa fa-check"></i><b>7</b> Ancestral Transcriptome Inference</a><ul>
<li class="chapter" data-level="7.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#theory-3"><i class="fa fa-check"></i><b>7.1</b> Theory</a><ul>
<li class="chapter" data-level="7.1.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#from-brownian-motion-bm-model-to-ornstein-uhlenbeck-ou-model"><i class="fa fa-check"></i><b>7.1.1</b> From Brownian Motion (BM) model to Ornstein-Uhlenbeck (OU) model</a></li>
<li class="chapter" data-level="7.1.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-ou-model-under-a-phylogeny"><i class="fa fa-check"></i><b>7.1.2</b> The OU model under a phylogeny</a></li>
<li class="chapter" data-level="7.1.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-gene-expression-inference-an-empirical-bayesian-approach"><i class="fa fa-check"></i><b>7.1.3</b> Ancestral gene expression inference: an empirical Bayesian approach</a></li>
<li class="chapter" data-level="7.1.4" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-problem-of-ancestral-transcriptome-inference"><i class="fa fa-check"></i><b>7.1.4</b> The problem of ancestral transcriptome inference</a></li>
<li class="chapter" data-level="7.1.5" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#stationary-ornstein-uhlenbeck-sou-model"><i class="fa fa-check"></i><b>7.1.5</b> Stationary Ornstein-Uhlenbeck (sOU) model</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#a-fast-algorithm-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>7.2</b> A Fast Algorithm under the Stationary OU Model</a></li>
<li class="chapter" data-level="7.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#case-study-ancestral-expression-inference-of-gene-emp1"><i class="fa fa-check"></i><b>7.3</b> Case Study: Ancestral Expression Inference of Gene <em>EMP1</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#expression-character-tree-1"><i class="fa fa-check"></i><b>7.3.1</b> Expression character tree</a></li>
<li class="chapter" data-level="7.3.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#creating-variance-co-variance-matrix"><i class="fa fa-check"></i><b>7.3.2</b> Creating variance co-variance matrix</a></li>
<li class="chapter" data-level="7.3.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-expression-estimation"><i class="fa fa-check"></i><b>7.3.3</b> Ancestral expression estimation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial: Use <em>TreeExp</em> for Phylogenetic Transcriptome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="estimating-the-strength-of-expression-conservation" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Estimating the Strength of Expression Conservation</h1>
<div id="theory-2" class="section level2">
<h2><span class="header-section-number">6.1</span> Theory</h2>
<div id="stabilizing-selection-model-in-transcriptome-evolution-across-species" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Stabilizing selection model in transcriptome evolution across species</h3>
<p>The Ornstein-Uhlenbeck (OU) model, which claims that expression changes are constrained by the stabilizing selection, is biologically more realistic than a simple Brownian motion (BM) model<span class="citation">(Rohlfs, Harrigan, and Nielsen <a href="#ref-rohlfs2014">2014</a> <span class="citation">Lemos et al. (<a href="#ref-lemos2005">2005</a>)</span><span class="citation">Gu and Su (<a href="#ref-gu2007">2007</a>)</span><span class="citation">Bedford and Hartl (<a href="#ref-bedford2009">2009</a>)</span><span class="citation">Brawand et al. (<a href="#ref-brawand2011">2011</a>)</span>)</span>.</p>
<p>Intuitively speaking, the OU model includes two opposite processes:</p>
<ul>
<li>random mutations push the expression level (<span class="math inline">\(x\)</span>) of a gene away from the optimum (<span class="math inline">\(\mu\)</span>), a process may suffer from a fitness reduction;</li>
<li>the return process to the expression optimum (<span class="math inline">\(\mu\)</span>) driven by a positive selection; the strength of this ‘elastic’ return increases proportionally with w, the coefficient of stabilizing selection. Given the initial expression value <span class="math inline">\({x_0}\)</span>, the OU model predicts that <span class="math inline">\(x\left(t\right)\)</span>, the expression level after <span class="math inline">\(t\)</span> evolutionary time units, follows a normal distribution with the mean <span class="math inline">\(E\left[x\  \mid\  x_0\right]\)</span> and variance <span class="math inline">\(V\left[x\  \mid\  x_0\right]\)</span></li>
</ul>
<p><span class="math display">\[
\begin{split}
E\left[x\  \mid\  x_0\right]\  &amp;=\  \mu\left(1-e^{-\beta t}\right)\  +\  z_0e^{-\beta t} \\
V\left[x\  \mid\  x_0\right]\  &amp;=\frac{1-e^{-2\beta t}}{W}
\end{split}\tag{3.1}
\]</span></p>
<p>respectively, where <span class="math inline">\(\beta\)</span> is the rate of expression evolution and W is the strength of expression conservation; symbolically one may write an OU process by <span class="math inline">\(OU\left(x\  \mid\  x_0;\  \theta\right)\)</span>, where <span class="math inline">\(\theta\  =\  \left(\mu,\  \beta t,\  W\right)\)</span> is the parameter vector.</p>
</div>
<div id="stationary-ou-model-under-a-species-phylogeny" class="section level3">
<h3><span class="header-section-number">6.1.2</span> Stationary OU model under a species phylogeny</h3>
<p>We shall develop a statistical method to estimate the strength (<span class="math inline">\(W\)</span>) of expression conservation of a gene from high throughput RNA-seq data of multiple species. <strong>Figure 6.1</strong> illustrates the evolutionary scenario used in the evolutionary transcriptome analysis.</p>
<ul>
<li>The first component is the evolutionary lineage from the origin of the tissue (node <span class="math inline">\(Z\)</span>) to the root (node <span class="math inline">\(O\)</span>) of the species phylogeny, with <span class="math inline">\(\tau\)</span> evolutionary time units.</li>
<li>The second component is the conventional species phylogeny with <span class="math inline">\(n\)</span> species. That is, given the initial expression value <span class="math inline">\(z_0\)</span> at node <span class="math inline">\(Z\)</span>, the OU process of <span class="math inline">\(x_0\)</span> in the lineage from <span class="math inline">\(Z\)</span> to <span class="math inline">\(O\)</span> is given by <span class="math inline">\(OU\left(x_0\mid z_0;\  \theta\right)\)</span>, where the parameter vector <span class="math inline">\(\theta=\left(\mu,\  \beta\tau,\  W\right)\)</span>. The joint density of expressions <span class="math inline">\(\boldsymbol{x}=\left(x_1,...x_n\right)\)</span> conditional of the expression level (<span class="math inline">\(x_0\)</span>) at root <span class="math inline">\(O\)</span>, denoted by <span class="math inline">\(P\left(\boldsymbol{x}\mid x_0\right)\)</span>, can be derived under the OU model. It follows that the joint expression density of <span class="math inline">\(\boldsymbol{x}=\left(x_1,...x_n\right)\)</span> conditional of <span class="math inline">\(z_0\)</span> is given by</li>
</ul>
<p><span class="math display">\[P\left(\boldsymbol{x}\mid z_0\right)=\int_{\  -\infty}^{\infty}OU\left(x_0\mid z_0;\  \tau\right)P\left(\boldsymbol{x}\mid x_0\right)dx_0\tag{3.2}\]</span>
<span class="citation">(Hansen and Martins <a href="#ref-hansen1996">1996</a>)</span> showed that either <span class="math inline">\(P\left(\boldsymbol{x}\mid x_0\right)\)</span> or <span class="math inline">\(P\left(\boldsymbol{x}\mid z_0\right)\)</span> is multivariate normally distributed. Consider <span class="math inline">\(P\left(\boldsymbol{x}\mid z_0\right)\sim N\left(\boldsymbol{x}; \boldsymbol{\mu}, \boldsymbol{V}\right)\)</span> at first, where <span class="math inline">\(\boldsymbol{\mu}\)</span> is the mean vector and <span class="math inline">\(\boldsymbol{V}\)</span> is the variance-covariance matrix. Because current transcriptome data contain little information about the evolution from node <span class="math inline">\(Z\)</span> to node <span class="math inline">\(O\)</span>, calculations of <span class="math inline">\(\boldsymbol{\mu}\)</span> and <span class="math inline">\(\boldsymbol{V}\)</span> are usually difficult because both depend on <span class="math inline">\(z_0\)</span> and <span class="math inline">\(\tau\)</span>.</p>
<p>We propose a stationary OU model (sOU) that helpful to avoid these problems in practice, which postulates that, at the genome-wide level, the biological function of a tissue-specific transcriptome is conservative during the course of species evolution. Specifically, sOU involves two assumptions.</p>
<ul>
<li>Origin of the tissue (node <span class="math inline">\(Z\)</span> in <strong>Figure 6.1</strong>) was so ancient that the evolutionary time between nodes <span class="math inline">\(Z\)</span> and <span class="math inline">\(O\)</span> can be approximated by <span class="math inline">\(\tau \to \infty\)</span>. Consequently, the expression mean and variance at root <span class="math inline">\(O\)</span> approach to <span class="math inline">\(\mu\)</span> and <span class="math inline">\(\rho^{2} = 1/W\)</span>, respectively.</li>
<li>The optimal expression level (<span class="math inline">\(\mu\)</span>) and the strength of expression conservation (<span class="math inline">\(W\)</span>) remain virtually constant along the species phylogeny, that is, the expression mean and variances at all internal and external nodes are equal to <span class="math inline">\(\mu\)</span> and <span class="math inline">\(1/W\)</span>, respectively.</li>
</ul>
<p>Under the stationary OU model, <span class="math inline">\(P\left(\boldsymbol{x}\mid z_0\right)\)</span> can be simplified as follows:</p>
<ul>
<li><span class="math inline">\(P\left(\boldsymbol{x}\mid z_0\right)\)</span> is independent of <span class="math inline">\(z_0\)</span>;</li>
<li>the mean vector <span class="math inline">\(\boldsymbol{\mu}\)</span> is uniform, i.e., <span class="math inline">\(\mu_1=...\mu_n\)</span>;</li>
<li>the variance-covariance matrix is simply given by <span class="math inline">\(\boldsymbol{V}=\boldsymbol{R}/W\)</span>, where <span class="math inline">\(\boldsymbol{R}\)</span> is the coefficient of correlation matrix. Our intent is to estimate the strength of expression conservation of a gene, characterized by a single parameter <span class="math inline">\(W\)</span>. In this sense, the joint density of <span class="math inline">\(\boldsymbol{x}\)</span> can be symbolically written by <span class="math inline">\(P\left(\boldsymbol{x};\boldsymbol{\mu},\boldsymbol{R},W\right)\)</span>. Together, we have</li>
</ul>
<p><span class="math display">\[P\left(\boldsymbol{x}\mid x_0\right)=P\left(\boldsymbol{x}\mid z_0\right)\sim P\left(\boldsymbol{x};\  \boldsymbol{\mu},\  \boldsymbol{R},\  W\right)=N\left(\boldsymbol{x};\  \boldsymbol{\mu},\  \boldsymbol{R}/W\right)\tag{3.3}\]</span></p>
<div class="figure"><span id="fig:pressure"></span>
<img src="Figure2.png" alt="The evolutionary scenario for comparative transcriptome analysis" width="50%" />
<p class="caption">
Figure 6.1: The evolutionary scenario for comparative transcriptome analysis
</p>
</div>
</div>
<div id="variation-of-w-among-genes" class="section level3">
<h3><span class="header-section-number">6.1.3</span> Variation of <span class="math inline">\(W\)</span> among genes</h3>
<p>The sOU model assumes that the strength of expression conservation (<span class="math inline">\(W\)</span>) of a gene remains a constant in species evolution but differs among genes. Substantial evidence has supported this argument<span class="citation">(Bedford and Hartl <a href="#ref-bedford2009">2009</a> <span class="citation">Brawand et al. (<a href="#ref-brawand2011">2011</a>)</span><span class="citation">Cui et al. (<a href="#ref-cui2007">2007</a>)</span><span class="citation">Park and Lehner (<a href="#ref-park2013">2013</a>)</span><span class="citation">Tirosh et al. (<a href="#ref-tirosh2006">2006</a>)</span><span class="citation">Warnefors and Eyre-Walker (<a href="#ref-warnefors2012">2012</a>)</span><span class="citation">Zou et al. (<a href="#ref-zou2011">2011</a>)</span>)</span>. Further, we model <span class="math inline">\(W\)</span> as a random variable that varies among genes according to a gamma distribution, that is,</p>
<p><span class="math display">\[\phi\left(W;\alpha,\overline{W}\right)=\frac{(\alpha/\overline{W})^\alpha}{\Gamma\left(\alpha\right)}W^{\alpha-1}e^{-\alpha W/\overline{W}}\tag{3.4}\]</span>
where <span class="math inline">\(\overline{W}\)</span> is the mean and <span class="math inline">\(\alpha\)</span> is the shape parameter; a small values of <span class="math inline">\(\alpha\)</span> means a high degree of <span class="math inline">\(W\)</span>-variation, and <span class="math inline">\(\alpha = \infty\)</span> means a constant <span class="math inline">\(W\)</span> among genes. After rewriting the joint normal density <span class="math inline">\(P\left(\boldsymbol{x};\  \boldsymbol{\mu},\  \boldsymbol{R},\  W\right)\)</span> by <span class="math inline">\(P\left(\boldsymbol{x}\mid W;\  \boldsymbol{\mu},\  \boldsymbol{R},\right)\)</span> to indicate <span class="math inline">\(W\)</span> is a random variable, we have
<span class="math display">\[P\left(\boldsymbol{x}\mid W;\  \boldsymbol{\mu},\  \boldsymbol{R},\right)=N\left(\boldsymbol{x};\  \boldsymbol{\mu},\  \boldsymbol{R}/W\right)\tag{3.5}\]</span>
It follows that the marginal density of <span class="math inline">\(\boldsymbol{x}\)</span> is given by
<span class="math display">\[
\begin{split}
P\left(\boldsymbol{x};\boldsymbol{\mu},\boldsymbol{R},\alpha,\overline{W}\right)&amp;=\int_0^{\infty}P\left(\boldsymbol{x}|W;\  \boldsymbol{\mu};\boldsymbol{R}\right)\phi\left(W;\  \alpha,\overline{W}\right)dW\\
&amp;=A\left(\frac{\overline{W}}{\alpha}\right)^{n/2}\left(\frac{\Gamma\left(n/2+\alpha\right)}{\Gamma\left(\alpha\right)}\right)\left(\frac{\alpha}{\alpha+Q\left(\boldsymbol{x}\right)\overline{W}}\right)^{n/2+\alpha}
\end{split}\tag{3.6}
\]</span>
where <span class="math inline">\(Q\left(\boldsymbol{x}\right)=\left(\boldsymbol{x}-\boldsymbol{\mu}\right)^T\boldsymbol{R}^{-1}\left(\boldsymbol{x}-\boldsymbol{\mu}\right)\)</span> is a quadratic function of <span class="math inline">\(\boldsymbol{x}\)</span>, and <span class="math inline">\(A=\pi^{-\frac{n}{2}}\left|\boldsymbol{R}\right|^{-\frac{1}{2}}\)</span> is a normalization constant.</p>
</div>
<div id="an-empirical-bayesian-framework-for-gene-specific-w-esitimation" class="section level3">
<h3><span class="header-section-number">6.1.4</span> An empirical Bayesian framework for gene-specific <span class="math inline">\(W\)</span> esitimation</h3>
<div id="posterior-mean-of-w-as-gene-specific-predictor" class="section level4">
<h4><span class="header-section-number">6.1.4.1</span> Posterior mean of W as gene-specific predictor</h4>
<p>We adopt an empirical Bayesian procedure to predict the strength of expression conservation for single gene. By the Bayes rule, the posterior density of <span class="math inline">\(W\)</span> conditional of the expression profile (<span class="math inline">\(\boldsymbol{x}\)</span>) of a gene is given by</p>
<p><span class="math display">\[P\left(W \mid \boldsymbol{x};\boldsymbol{\mu},\boldsymbol{R},\alpha,\overline{W} \right) =\frac{\phi\left(W;\alpha,\overline{W}\right)P\left(\boldsymbol{x}|W;\boldsymbol{\mu},\boldsymbol{R}\right)}{P\left(\boldsymbol{x};\boldsymbol{\mu},\boldsymbol{R},\alpha,\overline{W}\right)}\tag{3.7}\]</span>
After some mathematical calculations, one can show that the analytical form of the posterior density of <span class="math inline">\(W\)</span> is given by
<span class="math display">\[P\left(W \mid \boldsymbol{x};\boldsymbol{\mu},\boldsymbol{R},\alpha,\overline{W} \right) = A\left(\frac{\overline{W}}{\alpha}\right)^{n/2}\left(\frac{\Gamma\left(n/2+\alpha\right)}{\Gamma\left(\alpha\right)}\right)\left(\frac{\alpha}{\alpha+Q\left(\boldsymbol{x}\right)\overline{W}}\right)^{n/2+\alpha}\tag{3.8}\]</span>
Hence, <span class="math inline">\(P\left(W \mid \boldsymbol{x};\boldsymbol{\mu},\boldsymbol{R},\alpha,\overline{W} \right)\)</span> follows a gamma distribution, with the mean and variance given by
<span class="math display">\[
\begin{split}
E\left[W|\boldsymbol{x}\right]\  &amp;=\left[\frac{\alpha+\frac{n}{2}}{\alpha+Q\left(x\right)\overline{W}}\right]\overline{W}\\
Var\left[W|\boldsymbol{x}\right]&amp;=\left[\frac{\alpha+\frac{n}{2}}{\left(\alpha+Q\left(x\right)\overline{W}\right)^2}\right](\overline{W})^2
\end{split}\tag{3.9}
\]</span></p>
<p>respectively. Of particular, the posterior mean, <span class="math inline">\(E\left[W|\boldsymbol{x}\right]\)</span> can be used as the predictor for the strength of expression conservation of a gene with obseved expression profile <span class="math inline">\(\boldsymbol{x}\)</span>.</p>
</div>
<div id="relative-strength-of-expression-conservation" class="section level4">
<h4><span class="header-section-number">6.1.4.2</span> Relative strength of expression conservation</h4>
<p>It has been realized that the strenth of expression conservation (<span class="math inline">\(W_k\)</span>) of gene <span class="math inline">\(k\)</span> highly depends on the normalization method used for RNA-seq raw reads count. Hence, it is difficult to compare between two sets of estimates when they used different normalization methods. To aleviate this problem, it is more convenient to use the ratio <span class="math inline">\(U_k=W_k/\overline{W}\)</span>, the relative strength of expression conservation. Suppose we have <span class="math inline">\(N\)</span> orthologous genes under study, and the expression profile of the <span class="math inline">\(k\)</span>-th gene is denoted by <span class="math inline">\(\boldsymbol{x}_k\)</span>, <span class="math inline">\(k=1,…N\)</span>. Let <span class="math inline">\(W_{k}=E\left[W|\boldsymbol{x}\right]\)</span> be the posterior predictor for the strength of expression conservaton of gene <span class="math inline">\(k\)</span>. Notice that the expectation of the posterior mean prediction (<span class="math inline">\(E\left[W|\boldsymbol{x}\right]\)</span>) with respect to the the maginal density <span class="math inline">\(P\left(\boldsymbol{x};\boldsymbol{\mu},\boldsymbol{R},\alpha,\overline{W}\right)\)</span> is equal to the mean of the strength of expression conservation (<span class="math inline">\(\overline{W}\)</span>), that is,</p>
<p><span class="math display">\[\int E\left[W\mid \boldsymbol{x}\right]P\left(\boldsymbol{x};\  \boldsymbol{\mu},\  \boldsymbol{R},\  \alpha,\  \overline{W}\right)d\boldsymbol{x}=\overline{W}\tag{3.10}\]</span>
Eq.(3.10) implies that the averge of <span class="math inline">\(U_k\)</span>, the relative strength of expression conservation over all genes is roughly to be one, that is,<br />
<span class="math display">\[\frac{\sum_{k=1}^NU_k}{N}\approx1\tag{3.11}\]</span>
</p>
</div>
</div>
</div>
<div id="statistical-procedure" class="section level2">
<h2><span class="header-section-number">6.2</span> Statistical Procedure</h2>
<p>Suppose that we have RNA-seq datasets of a particular tissue from <span class="math inline">\(n\)</span> species, and the expression profile of each <span class="math inline">\(k\)</span>-th gene denoted by <span class="math inline">\(\boldsymbol{x_k}=\left(x_{1k,\  }...x_{nk}\right)\)</span>, <span class="math inline">\(k=1...,N\)</span>. We developed a practically feasible procedure to estimate <span class="math inline">\(W\)</span> of each gene, which actually deals with the quadratic function of <span class="math inline">\(\boldsymbol{x_{k}}\)</span>, or <span class="math inline">\(Q\left(\boldsymbol{x_k}\right)\)</span>. The procedure is briefly described below.</p>
<ol style="list-style-type: decimal">
<li>Calculate gene-<span class="math inline">\(k\)</span> specific mean (<span class="math inline">\(\mu_{k}\)</span>) by a simple average over orthologous genes.</li>
<li>Calculate the matrix of correlation coefficients (<span class="math inline">\(\boldsymbol{R}\)</span>) from comparative RNA-seq data, which is applied to each of gene.</li>
<li>Calculate the quadratic function of each gene <span class="math inline">\(k\)</span>, <span class="math inline">\(Q\left(x_k\right)\)</span>, by
<span class="math display">\[\hat{Q}\left(x_k\right)=\sum_{_{i=1}}^n\sum_{j=1}^nc_{ij}\left(x_{ik}-\mu_k\right)\left(x_{jk}-\mu_k\right)\tag{3.12}\]</span>
where <span class="math inline">\(c_{ij}\)</span> is the <span class="math inline">\(ij\)</span>-th element of matrix <span class="math inline">\(\boldsymbol{C}=\boldsymbol{R}^{-1}\)</span>, <span class="math inline">\(x_{ik}\)</span> or <span class="math inline">\(x_{jk}\)</span> is the expression value of gene <span class="math inline">\(k\)</span> in species <span class="math inline">\(i\)</span> or <span class="math inline">\(j\)</span>, respectively.</li>
<li>Treating <span class="math inline">\(Q\left(\boldsymbol{x_k}\right)\)</span> as the observation of gene <span class="math inline">\(k\)</span> and rewriting Eq.(6), symbolically, by <span class="math inline">\(P\left(Q\left(\boldsymbol{x_k}\right);\alpha,\  \overline{W}\right)\)</span>, build up an approximate likelihood function
<span class="math display">\[\ell\left(X \  \mid \  \alpha,\  \overline{W}\right)\  =  \prod_{k=1}^NP\left(\hat{Q}\left(\boldsymbol{x_k}\right);\  \alpha,\overline{W}\right)\]</span>
and obtain the maximum likelihood estimates (MLE) of <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\overline{W}\)</span>; the standard likelihood ratio test is applied to test the null hypothesis of no <span class="math inline">\(W\)</span>-variation among genes, i.e., <span class="math inline">\(\alpha = \infty\)</span>.</li>
<li>Calculate <span class="math inline">\(W_{k}\)</span>, the empirical Bayesian estimate of the strength of expression conservation of gene <span class="math inline">\(k\)</span>, from Eq.(3.9) after replacing <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\overline{W}\)</span> by their estimates.</li>
</ol>

</div>
<div id="case-study-emprical-bayesian-esitimates-of-expression-conservation-of-genes" class="section level2">
<h2><span class="header-section-number">6.3</span> Case Study: Emprical Bayesian Esitimates of Expression Conservation of genes</h2>
<p>We use the expression values of 5635 1:1 orthologous genes in brain of nine mammalian species to estimate the parameters of the selection pressure gamma distribution in brain. Then we estimate the gene-specific selection pressure based on Bayes’ theorem.</p>
<p><em>TreeExp</em> can be loaded the package in the usual way:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&#39;TreeExp&#39;</span>)</code></pre>
<p>Let us first load the tetrapod expression dataset:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&#39;tetraExp&#39;</span>)</code></pre>
<div id="inversed-correlation-matrix" class="section level3">
<h3><span class="header-section-number">6.3.1</span> Inversed correlation matrix</h3>
<p>And then, based on the constructed <em>taxaExp</em> object, we are going to create an inverse correlation matrix between mammalian species from the <em>taxaExp</em> object:</p>
<pre class="sourceCode r"><code class="sourceCode r">species.group &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Human&quot;</span>, <span class="st">&quot;Chimpanzee&quot;</span>, <span class="st">&quot;Bonobo&quot;</span>, <span class="st">&quot;Gorilla&quot;</span>,
<span class="st">&quot;Macaque&quot;</span>, <span class="st">&quot;Mouse&quot;</span>, <span class="st">&quot;Opossum&quot;</span>, <span class="st">&quot;Platypus&quot;</span>)
### all mammalian species

inv.corr.mat &lt;-<span class="st"> </span><span class="kw">corrMatInv</span>(tetraExp, <span class="dt">taxa =</span> species.group, <span class="dt">subtaxa =</span> <span class="st">&quot;Brain&quot;</span>)
inv.corr.mat</code></pre>
<pre><code>##                  Human_Brain Chimpanzee_Brain Bonobo_Brain Gorilla_Brain
## Human_Brain       15.7867153      -7.05168768   -0.5467655   -5.21892601
## Chimpanzee_Brain  -7.0516877      26.74465629  -13.4479799   -4.73381009
## Bonobo_Brain      -0.5467655     -13.44797986   18.3508826   -1.88760497
## Gorilla_Brain     -5.2189260      -4.73381009   -1.8876050   14.49822033
## Macaque_Brain     -1.9749665      -1.07677087   -1.3431113   -2.65851421
## Mouse_Brain       -0.6505556      -0.21643982   -0.3682007    0.27736989
## Opossum_Brain     -0.1345412      -0.31082672   -0.1962514   -0.02468375
## Platypus_Brain    -0.1113688      -0.01113911   -0.4318870   -0.01712743
##                  Macaque_Brain Mouse_Brain Opossum_Brain Platypus_Brain
## Human_Brain         -1.9749665  -0.6505556   -0.13454125    -0.11136876
## Chimpanzee_Brain    -1.0767709  -0.2164398   -0.31082672    -0.01113911
## Bonobo_Brain        -1.3431113  -0.3682007   -0.19625137    -0.43188695
## Gorilla_Brain       -2.6585142   0.2773699   -0.02468375    -0.01712743
## Macaque_Brain        9.9086664  -2.0792122   -0.50552772    -0.26332694
## Mouse_Brain         -2.0792122   5.1274351   -1.27896910    -0.61687114
## Opossum_Brain       -0.5055277  -1.2789691    3.90792798    -1.19664014
## Platypus_Brain      -0.2633269  -0.6168711   -1.19664014     3.02761767</code></pre>
</div>
<div id="estimation-of-gamma-parameters" class="section level3">
<h3><span class="header-section-number">6.3.2</span> Estimation of gamma parameters</h3>
<p>Then we need to extract the expression values of orthologous genes from the <em>taxaExp</em> object using <code>exptabTE</code> function.</p>
<pre class="sourceCode r"><code class="sourceCode r">brain.exptable &lt;-<span class="st"> </span><span class="kw">exptabTE</span>(tetraExp, <span class="dt">taxa =</span> species.group, <span class="dt">subtaxa =</span> <span class="st">&quot;Brain&quot;</span> ,<span class="dt">logrithm =</span> <span class="ot">TRUE</span>)
<span class="kw">head</span>(brain.exptable)</code></pre>
<pre><code>##                 Human_Brain Chimpanzee_Brain Bonobo_Brain Gorilla_Brain
## ENSG00000198824   4.6111724         5.029011     5.151778     4.1093606
## ENSG00000118402   5.3652726         5.597233     5.360715     4.9064096
## ENSG00000166167   6.6794801         6.687621     7.174127     6.4891254
## ENSG00000144724   4.8374387         4.356144     5.460087     4.2280490
## ENSG00000183508   0.9855004         1.327687     2.788686     0.9030383
## ENSG00000008086   4.9321557         4.516015     5.825277     4.7213727
##                 Macaque_Brain Mouse_Brain Opossum_Brain Platypus_Brain
## ENSG00000198824      5.861707    6.700994      7.218781       6.536675
## ENSG00000118402      5.909533    7.396091      7.014913       8.321252
## ENSG00000166167      7.304967    8.257011      7.593279       6.903038
## ENSG00000144724      5.852249    6.657211      7.164605       6.239551
## ENSG00000183508      1.395063    3.382667      3.240314       2.989139
## ENSG00000008086      7.000113    6.889352      7.595966       7.036943</code></pre>
<p>With the inverse correlation matrix and expression values of brain tissue in 9 mammals, we are now able to estimate the parameters of the gamma distribution:</p>
<pre class="sourceCode r"><code class="sourceCode r">gamma.paras &lt;-<span class="st"> </span><span class="kw">estParaGamma</span>(<span class="dt">exptable =</span>brain.exptable, <span class="dt">corrmatinv =</span>inv.corr.mat)
## print the elements of gamma.paras
gamma.paras</code></pre>
<pre><code>## $alpha
## [1] 2.937128
## 
## $W_average
## [1] 0.2610562
## 
## $speNum
## [1] 8
## 
## $geneNum
## [1] 5636</code></pre>
<p>The <span class="math inline">\(\bar{W}\)</span> is the average of the selection pressure levels in the tissue brain.
And the shape parameter <span class="math inline">\(\alpha\)</span> here can reflect the internal variances of selection pressure. The more close <span class="math inline">\(\alpha\)</span> is to 2, the more distinctive selection pressures on genes. And if the <span class="math inline">\(\alpha\)</span> is close to infinite, it means there are no difference among selection pressures on genes.</p>
</div>
<div id="bayesian-estimation-of-gene-specific-selection-pressure" class="section level3">
<h3><span class="header-section-number">6.3.3</span> Bayesian estimation of gene-specific selection pressure</h3>
<p>After parameters of the gamma distribution are estimated, we are able to estimate posterior selection pressures as well as their <em>se</em> with given ‘RPKM’ values across species:</p>
<pre class="sourceCode r"><code class="sourceCode r">brain.Q &lt;-<span class="st"> </span><span class="kw">estParaQ</span>(brain.exptable, <span class="dt">corrmatinv =</span> inv.corr.mat)
<span class="co"># with prior expression values and inversed correlation matrix</span>
    
brain.post&lt;-<span class="st"> </span><span class="kw">estParaWBayesian</span>(brain.Q, gamma.paras)
brain.W &lt;-<span class="st"> </span>brain.post<span class="op">$</span>w <span class="co"># posterior selection pressures</span>
brain.CI &lt;-<span class="st"> </span>brain.post<span class="op">$</span>ci95 <span class="co"># posterior expression 95% confidence interval</span></code></pre>
<p>After esitimating the Bayesian selection pressres, <span class="math inline">\(W\)</span>, we can chek a few gene with hightest selection pressure.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(brain.W) &lt;-<span class="st"> </span><span class="kw">rownames</span>(brain.exptable)
<span class="kw">head</span>(<span class="kw">sort</span>(brain.W, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)) <span class="co">#check a few genes with highest seletion pressure</span></code></pre>
<pre><code>## ENSG00000137270 ENSG00000102243 ENSG00000139515 ENSG00000146378 
##        0.616582        0.616582        0.616582        0.616582 
## ENSG00000151379 ENSG00000111049 
##        0.616582        0.616582</code></pre>
<p>and draw the dentisty plot of <span class="math inline">\(W\)</span> among genes.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">density</span>(brain.W))</code></pre>
<p><img src="TreeExp-Tutorial_files/figure-html/unnamed-chunk-33-1.png" width="576" /></p>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-bedford2009">
<p>Bedford, Trevor, and Daniel L. Hartl. 2009. “Optimization of gene expression by natural selection.” <em>Proceedings of the National Academy of Sciences</em> 106:1133–8. <a href="https://doi.org/10.1073/pnas.0812009106" class="uri">https://doi.org/10.1073/pnas.0812009106</a>.</p>
</div>
<div id="ref-brawand2011">
<p>Brawand, David, Magali Soumillon, Anamaria Necsulea, Philippe Julien, Gábor Csárdi, Patrick Harrigan, Manuela Weier, et al. 2011. “The evolution of gene expression levels in mammalian organs.” <em>Nature</em> 478:343. <a href="https://doi.org/10.1038/nature10532" class="uri">https://doi.org/10.1038/nature10532</a>.</p>
</div>
<div id="ref-cui2007">
<p>Cui, Qinghua, Zhenbao Yu, Enrico O. Purisima, and Edwin Wang. 2007. “MicroRNA regulation and interspecific variation of gene expression.” <em>Trends in Genetics</em> 23:372–75. <a href="https://doi.org/10.1016/j.tig.2007.04.003" class="uri">https://doi.org/10.1016/j.tig.2007.04.003</a>.</p>
</div>
<div id="ref-gu2007">
<p>Gu, Xun, and Zhixi Su. 2007. “Tissue-driven hypothesis of genomic evolution and sequence-expression correlations.” <em>Proceedings of the National Academy of Sciences</em> 104:2779–84. <a href="https://doi.org/10.1073/pnas.0610797104" class="uri">https://doi.org/10.1073/pnas.0610797104</a>.</p>
</div>
<div id="ref-hansen1996">
<p>Hansen, Thomas F, and Emilia P Martins. 1996. “Translating Between Microevolutionary Process and Macroevolutionary Patterns: The Correlation Structure of Interspecific Data.” <em>Evolution</em> 50:1404. <a href="https://doi.org/10.2307/2410878" class="uri">https://doi.org/10.2307/2410878</a>.</p>
</div>
<div id="ref-lemos2005">
<p>Lemos, Bernardo, Colin D. Meiklejohn, Mario Cáceres, and Daniel L. Hartl. 2005. “Rate of divergence in gene expression profiles of primates, mice and flies: stabilizing selection and variability among functionnal categories.” <em>Evolution</em> 63:126. <a href="https://doi.org/10.1554/04-251" class="uri">https://doi.org/10.1554/04-251</a>.</p>
</div>
<div id="ref-park2013">
<p>Park, Solip, and Ben Lehner. 2013. “Epigenetic epistatic interactions constrain the evolution of gene expression.” <em>Molecular Systems Biology</em> 9:645. <a href="https://doi.org/10.1038/msb.2013.2" class="uri">https://doi.org/10.1038/msb.2013.2</a>.</p>
</div>
<div id="ref-rohlfs2014">
<p>Rohlfs, Rori V., Patrick Harrigan, and Rasmus Nielsen. 2014. “Modeling Gene Expression Evolution with an Extended Ornstein–Uhlenbeck Process Accounting for Within-Species Variation.” <em>Molecular Biology and Evolution</em> 31:201–11. <a href="https://doi.org/10.1093/molbev/mst190" class="uri">https://doi.org/10.1093/molbev/mst190</a>.</p>
</div>
<div id="ref-tirosh2006">
<p>Tirosh, Itay, Adina Weinberger, Miri Carmi, and Naama Barkai. 2006. “A genetic signature of interspecies variations in gene expression.” <em>Nature Genetics</em> 38:ng1819. <a href="https://doi.org/10.1038/ng1819" class="uri">https://doi.org/10.1038/ng1819</a>.</p>
</div>
<div id="ref-warnefors2012">
<p>Warnefors, Maria, and Adam Eyre-Walker. 2012. “A Selection Index for Gene Expression Evolution and Its Application to the Divergence between Humans and Chimpanzees.” <em>PLoS ONE</em> 7:e34935. <a href="https://doi.org/10.1371/journal.pone.0034935" class="uri">https://doi.org/10.1371/journal.pone.0034935</a>.</p>
</div>
<div id="ref-zou2011">
<p>Zou, Yangyun, Wei Huang, Zhenglong Gu, and Xun Gu. 2011. “Predominant Gain of Promoter TATA Box after Gene Duplication Associated with Stress Responses.” <em>Molecular Biology and Evolution</em> 28:2893–2904. <a href="https://doi.org/10.1093/molbev/msr116" class="uri">https://doi.org/10.1093/molbev/msr116</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="relative-rate-test-for-transcriptome-evolution.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="ancestral-transcriptome-inference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05-Conservation.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["TreeExp-Tutorial.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
