<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Pairwise Expression Distance and Expression Tree-Making | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.10 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Pairwise Expression Distance and Expression Tree-Making | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Pairwise Expression Distance and Expression Tree-Making | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Jingwen Yang" />


<meta name="date" content="2019-05-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-manipulation-and-storage.html">
<link rel="next" href="relative-rate-test-for-transcriptome-evolution.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"><i>TreeExp</i>-Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#scope"><i class="fa fa-check"></i><b>2.1</b> Scope</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#citation"><i class="fa fa-check"></i><b>2.3</b> Citation</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#how-to-get-help"><i class="fa fa-check"></i><b>2.4</b> How to Get Help?</a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#rna-seq-data-enbeded"><i class="fa fa-check"></i><b>2.5</b> RNA-seq Data Enbeded</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html"><i class="fa fa-check"></i><b>3</b> Data Manipulation and Storage</a><ul>
<li class="chapter" data-level="3.1" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#input-format"><i class="fa fa-check"></i><b>3.1</b> Input Format</a></li>
<li class="chapter" data-level="3.2" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#example-of-input-data"><i class="fa fa-check"></i><b>3.2</b> Example of Input Data</a></li>
<li class="chapter" data-level="3.3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#construction"><i class="fa fa-check"></i><b>3.3</b> Construction</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html"><i class="fa fa-check"></i><b>4</b> Pairwise Expression Distance and Expression Tree-Making</a><ul>
<li class="chapter" data-level="4.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#theory"><i class="fa fa-check"></i><b>4.1</b> Theory</a><ul>
<li class="chapter" data-level="4.1.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-ornstein-uhlenbeck-ou-model-of-transcriptome-evolution"><i class="fa fa-check"></i><b>4.1.1</b> The Ornstein-Uhlenbeck (OU) model of transcriptome evolution</a></li>
<li class="chapter" data-level="4.1.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#transcriptome-evolution-between-species"><i class="fa fa-check"></i><b>4.1.2</b> Transcriptome evolution between species</a></li>
<li class="chapter" data-level="4.1.3" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-estimation-problem-of-expression-distance"><i class="fa fa-check"></i><b>4.1.3</b> The estimation problem of expression distance</a></li>
<li class="chapter" data-level="4.1.4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-expression-distance-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>4.1.4</b> Estimation of expression distance under the stationary OU model</a></li>
<li class="chapter" data-level="4.1.5" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-distance-estimation-when-expression-optima-vary-among-genes"><i class="fa fa-check"></i><b>4.1.5</b> Expression distance estimation when expression optima vary among genes</a></li>
<li class="chapter" data-level="4.1.6" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-pi"><i class="fa fa-check"></i><b>4.1.6</b> Estimation of <span class="math inline">\(\pi\)</span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#case-study-expression-tree-building"><i class="fa fa-check"></i><b>4.2</b> Case Study: Expression Tree Building</a><ul>
<li class="chapter" data-level="4.2.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#distance-matrix"><i class="fa fa-check"></i><b>4.2.1</b> Distance matrix</a></li>
<li class="chapter" data-level="4.2.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-character-tree"><i class="fa fa-check"></i><b>4.2.2</b> Expression character tree</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html"><i class="fa fa-check"></i><b>5</b> Relative Rate Test for Transcriptome Evolution</a><ul>
<li class="chapter" data-level="5.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#theory-1"><i class="fa fa-check"></i><b>5.1</b> Theory</a></li>
<li class="chapter" data-level="5.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#statistical-procedure-of-the-relative-rate-test"><i class="fa fa-check"></i><b>5.2</b> Statistical Procedure of the Relative Rate Test</a><ul>
<li class="chapter" data-level="5.2.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#sampling-variance-of-the-expression-distance"><i class="fa fa-check"></i><b>5.2.1</b> Sampling variance of the expression distance</a></li>
<li class="chapter" data-level="5.2.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#calculation-of-varbigtriangleup_ab"><i class="fa fa-check"></i><b>5.2.2</b> Calculation of <span class="math inline">\(Var(\bigtriangleup_{AB})\)</span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#case-study-testing-fast-evolving-genes-in-human"><i class="fa fa-check"></i><b>5.3</b> Case Study: Testing Fast-evolving Genes in Human</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html"><i class="fa fa-check"></i><b>6</b> Estimating the Strength of Expression Conservation</a><ul>
<li class="chapter" data-level="6.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#theory-2"><i class="fa fa-check"></i><b>6.1</b> Theory</a><ul>
<li class="chapter" data-level="6.1.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stabilizing-selection-model-in-transcriptome-evolution-across-species"><i class="fa fa-check"></i><b>6.1.1</b> Stabilizing selection model in transcriptome evolution across species</a></li>
<li class="chapter" data-level="6.1.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stationary-ou-model-under-a-species-phylogeny"><i class="fa fa-check"></i><b>6.1.2</b> Stationary OU model under a species phylogeny</a></li>
<li class="chapter" data-level="6.1.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#variation-of-w-among-genes"><i class="fa fa-check"></i><b>6.1.3</b> Variation of <span class="math inline">\(W\)</span> among genes</a></li>
<li class="chapter" data-level="6.1.4" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#an-empirical-bayesian-framework-for-gene-specific-w-esitimation"><i class="fa fa-check"></i><b>6.1.4</b> An empirical Bayesian framework for gene-specific <span class="math inline">\(W\)</span> esitimation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#statistical-procedure"><i class="fa fa-check"></i><b>6.2</b> Statistical Procedure</a></li>
<li class="chapter" data-level="6.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#case-study-emprical-bayesian-esitimates-of-expression-conservation-of-genes"><i class="fa fa-check"></i><b>6.3</b> Case Study: Emprical Bayesian Esitimates of Expression Conservation of genes</a><ul>
<li class="chapter" data-level="6.3.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#inversed-correlation-matrix"><i class="fa fa-check"></i><b>6.3.1</b> Inversed correlation matrix</a></li>
<li class="chapter" data-level="6.3.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#estimation-of-gamma-parameters"><i class="fa fa-check"></i><b>6.3.2</b> Estimation of gamma parameters</a></li>
<li class="chapter" data-level="6.3.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#bayesian-estimation-of-gene-specific-selection-pressure"><i class="fa fa-check"></i><b>6.3.3</b> Bayesian estimation of gene-specific selection pressure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html"><i class="fa fa-check"></i><b>7</b> Ancestral Transcriptome Inference</a><ul>
<li class="chapter" data-level="7.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#theory-3"><i class="fa fa-check"></i><b>7.1</b> Theory</a><ul>
<li class="chapter" data-level="7.1.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#from-brownian-motion-bm-model-to-ornstein-uhlenbeck-ou-model"><i class="fa fa-check"></i><b>7.1.1</b> From Brownian Motion (BM) model to Ornstein-Uhlenbeck (OU) model</a></li>
<li class="chapter" data-level="7.1.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-ou-model-under-a-phylogeny"><i class="fa fa-check"></i><b>7.1.2</b> The OU model under a phylogeny</a></li>
<li class="chapter" data-level="7.1.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-gene-expression-inference-an-empirical-bayesian-approach"><i class="fa fa-check"></i><b>7.1.3</b> Ancestral gene expression inference: an empirical Bayesian approach</a></li>
<li class="chapter" data-level="7.1.4" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-problem-of-ancestral-transcriptome-inference"><i class="fa fa-check"></i><b>7.1.4</b> The problem of ancestral transcriptome inference</a></li>
<li class="chapter" data-level="7.1.5" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#stationary-ornstein-uhlenbeck-sou-model"><i class="fa fa-check"></i><b>7.1.5</b> Stationary Ornstein-Uhlenbeck (sOU) model</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#a-fast-algorithm-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>7.2</b> A Fast Algorithm under the Stationary OU Model</a></li>
<li class="chapter" data-level="7.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#case-study-ancestral-expression-inference-of-gene-emp1"><i class="fa fa-check"></i><b>7.3</b> Case Study: Ancestral Expression Inference of Gene <em>EMP1</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#expression-character-tree-1"><i class="fa fa-check"></i><b>7.3.1</b> Expression character tree</a></li>
<li class="chapter" data-level="7.3.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#creating-variance-co-variance-matrix"><i class="fa fa-check"></i><b>7.3.2</b> Creating variance co-variance matrix</a></li>
<li class="chapter" data-level="7.3.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-expression-estimation"><i class="fa fa-check"></i><b>7.3.3</b> Ancestral expression estimation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/jingwyang/TreeExp" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial: Use <em>TreeExp</em> for Phylogenetic Transcriptome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pairwise-expression-distance-and-expression-tree-making" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Pairwise Expression Distance and Expression Tree-Making</h1>
<div id="theory" class="section level2">
<h2><span class="header-section-number">4.1</span> Theory</h2>
<div id="the-ornstein-uhlenbeck-ou-model-of-transcriptome-evolution" class="section level3">
<h3><span class="header-section-number">4.1.1</span> The Ornstein-Uhlenbeck (OU) model of transcriptome evolution</h3>
<p>It is generally believed that the expression level of a gene, denoted by <span class="math inline">\(x\)</span>, is subject to the stabilizing selection to maintain the optimum during the course of evolution. Consequently, the motion of the expression level (<span class="math inline">\(x\)</span>) can be described by the so-called Ornstein-Uhlenbeck (OU) process<span class="citation">(Hansen and Martins <a href="#ref-hansen1996">1996</a> <span class="citation">Bergmann, Ihmels, and Barkai (<a href="#ref-bergmann2004">2004</a>)</span><span class="citation">Butler and King (<a href="#ref-butler2004">2004</a>)</span>)</span>. Briefing speaking, while the random force driving the expression level away from the optimum, the deterministic force will pull the expression level back to the optimum. It has been shown that the distribution density of <span class="math inline">\(x\)</span> after <span class="math inline">\(t\)</span> time units since the initial value <span class="math inline">\(x_0\)</span>, denoted by <span class="math inline">\(OU\left(x\mid x_0,\theta\right)\)</span>, is normal with the mean <span class="math inline">\(E\left[x\mid x_0\right]\)</span>and variance <span class="math inline">\(V\left(x\mid x_0\right)\)</span> given by</p>
<p><span class="math display">\[
\begin{split}
E\left[x\  \mid\  x_0\right]\  &amp;=\  \mu\left(1-e^{-\beta t}\right)\  +\  z_0e^{-\beta t} \\
V\left[x\  \mid\  x_0\right]\  &amp;=\frac{1-e^{-2\beta t}}{W}
\end{split}\tag{1.1}
\]</span></p>
<p>respectively, where <span class="math inline">\(\theta\)</span> is for the parameter vector: <span class="math inline">\(\mu\)</span> is the optimal expression value, <span class="math inline">\(\beta\)</span> the rate of expression evolution, and <span class="math inline">\(W\)</span> the strength of expression conservation.</p>
</div>
<div id="transcriptome-evolution-between-species" class="section level3">
<h3><span class="header-section-number">4.1.2</span> Transcriptome evolution between species</h3>
<p>Consider a simple two-stage scenario of transcriptome evolution that consists of two species that diverged <span class="math inline">\(t\)</span> time units ago (<strong>Figure 1A</strong>).</p>
<div class="figure"><span id="fig:unnamed-chunk-14"></span>
<img src="Figure1.png" alt="Figure1. The schematic of transcriptome evolution along a phylogeny" width="90%" />
<p class="caption">
Figure 4.1: Figure1. The schematic of transcriptome evolution along a phylogeny
</p>
</div>
<p>The first stage is the evolutionary lineage from the tissue ancestor (denoted by node <span class="math inline">\(Z\)</span>) to the common ancestor of two species (denoted by node <span class="math inline">\(O\)</span>), with a span of <span class="math inline">\(\tau\)</span> evolutionary time units. Given the initial expression value <span class="math inline">\(z_0\)</span> at node <span class="math inline">\(Z\)</span>, the density of <span class="math inline">\(x_0\)</span> (the ancestral expression level at node <span class="math inline">\(O\)</span>) is given by <span class="math inline">\(OU\left(x\mid x_0,\  \theta_0\right)\)</span> where the parameter vector <span class="math inline">\(\theta_{0}=\left(\mu,\  \beta\tau,\  W\right)\)</span>.</p>
<p>In the second stage, let <span class="math inline">\(x_1\)</span> and <span class="math inline">\(x_2\)</span> be the expression levels of an orthologous gene pair, respectively. Given the ancestral expression level (<span class="math inline">\(x_0\)</span>) at node <span class="math inline">\(O\)</span>, the density of <span class="math inline">\(x_1\)</span> follows <span class="math inline">\(OU\left(x_1\mid x_0;\  \theta_{1}\right)\)</span>, and that of <span class="math inline">\(x_2\)</span> follows <span class="math inline">\(OU\left(x_2\mid x_0;\  \theta_{2}\right)\)</span>, respectively, where <span class="math inline">\(\theta_{1}=\left(\mu,\  \beta_{1}t,\  W_{1}\right)\)</span> and <span class="math inline">\(\theta_{2}=\left(\mu,\  \beta_{2}t,\  W_{2}\right)\)</span>. If transcriptome evolution is independent between lineages, the joint density of <span class="math inline">\(x_1\)</span> and <span class="math inline">\(x_2\)</span> conditional of <span class="math inline">\(x_0\)</span> is simply given by <span class="math inline">\(P\left(x_1,\  x_2\  \mid\  x_0\right)=OU\left(x_1\mid x_0;\  \theta_1\right)OU\left(x_2\mid\  x_0;\  \theta_2\right)\)</span>. Since <span class="math inline">\(x_0\)</span> follows the density <span class="math inline">\(\pi\left(x_0\right)=OU\left(x_0\mid z_0;\  \theta_{0}\right)\)</span>, one can show that the joint density of <span class="math inline">\(x_1\)</span> and <span class="math inline">\(x_2\)</span></p>
<p><span class="math display">\[P\left(x_1,x_2\right)=\int_{-\infty}^{\infty}P\left(x_1,x_2\mid x_0\right)\pi\left(x_0\right)dx_0\tag{1.2}\]</span>
follows a normal distribution, with the mean vector
<span class="math display">\[
\begin{split}
E\left[x_1\right]&amp;=\mu\left(1-e^{\beta_1t}\right)+\left[\mu\left(1-e^{\beta_0\tau}\right)+z_0e^{\beta_0\tau}\right]e^{-\beta_1t} \\
E\left[x_2\right]&amp;=\mu\left(1-e^{\beta_2t}\right)+\left[\mu\left(1-e^{\beta_0\tau}\right)+z_0e^{\beta_0\tau}\right]e^{-\beta_2t} 
\end{split}\tag{1.3}
\]</span></p>
<p>and the variance-covariance matrix <span class="math inline">\(\boldsymbol{V}\)</span>given by</p>
<p><span class="math display">\[
\begin{split}
V_{11}&amp;=\frac{1}{W_1}+\left(\rho^2-\frac{1}{W_1}\right)e^{-\beta_1t}\\
V_{22}&amp;=\frac{1}{W_2}+\left(\rho^2-\frac{1}{W_2}\right)e^{-\beta_2t}\\
V_{12}&amp;=\rho^2e^{-\left(\beta_1+\beta_2\right)t}
\end{split}
\tag{1.4}
\]</span></p>
<p>where <span class="math inline">\(\rho^2=V\left(x_0\mid\  z\right)=\frac{\left(1-e^{-\beta\tau}\right)}{W}\)</span> is the variance of <span class="math inline">\(x_0\)</span> at root <span class="math inline">\(O\)</span>.</p>
</div>
<div id="the-estimation-problem-of-expression-distance" class="section level3">
<h3><span class="header-section-number">4.1.3</span> The estimation problem of expression distance</h3>
<p>Consider two RNA-seq datasets of the same tissue from species-1 and species-2, respectively, which include <span class="math inline">\(N\)</span> orthologous genes. For the <span class="math inline">\(k\)</span>-th orthologous pair, let <span class="math inline">\(x_{k,1}\)</span> and <span class="math inline">\(x_{k,2}\)</span> be the expression levels, respectively,<span class="math inline">\(k=1,\cdots,N\)</span>. It is straightforward to calculate the expression variances and their covariance, denoted by <span class="math inline">\(Var(X_1)\)</span>, <span class="math inline">\(Var(X_2)\)</span> and <span class="math inline">\(Cov(X_1, X_2)\)</span>, respectively. Meanwhile, the last equation of Eq.(1.4) shows that, theoretically, the expression covariance between species (<span class="math inline">\(V_{12})\)</span> decays exponentially with the component (<span class="math inline">\(\beta_1+\beta_2\)</span>), providing a foundation to define the expression distance between two species by
<span class="math display">\[D_{12}=\left(\beta_1+\beta_2\right)t=2\beta t\tag{1.5}\]</span>
where <span class="math inline">\(\beta=\frac{\left(\beta_1+\beta_2\right)}{2}\)</span> is the rate of expression evolution (averaged over two lineages). However, the difficulty to estimate <span class="math inline">\(D_{12}\)</span> is two-folds</p>
<ul>
<li>the ancient variance (<span class="math inline">\(\rho^2\)</span>) at root <span class="math inline">\(O\)</span> is usually unknown.</li>
<li>equations in Eq.(4) are valid only when the optimal expression level (<span class="math inline">\(\mu\)</span>) is constant among genes. We address these issues as follows.</li>
</ul>
</div>
<div id="estimation-of-expression-distance-under-the-stationary-ou-model" class="section level3">
<h3><span class="header-section-number">4.1.4</span> Estimation of expression distance under the stationary OU model</h3>
<p>We invoke the stationary OU process (sOU) to estimate the unknown ancestral variance <span class="math inline">\(\rho^2\)</span>. It has two assumptions:</p>
<ol style="list-style-type: decimal">
<li>the evolutionary span (<span class="math inline">\(\tau\)</span>) between the tissue ancestor (<span class="math inline">\(Z\)</span>) and the species ancestor (<span class="math inline">\(O\)</span>) is so large that the variance at root <span class="math inline">\(O\)</span> approaches to <span class="math inline">\(\rho^2=1/W\)</span>;</li>
<li>after the speciation, the strength of expression conservation (<span class="math inline">\(W\)</span>) remain constant between species. Under the constant-<span class="math inline">\(\mu\)</span> assumption, from Eqs.(1.3) and (1.4) we then have <span class="math inline">\(E\left[x_1\right]=E\left[x_2\right]=\mu\)</span>, <span class="math inline">\(V_{11}=V_{22}=\frac{1}{W}\)</span>, as well as the last equation of Eq.(1.4) by
<span class="math display">\[V_{12}=\frac{e^{-2\beta t}}{W}\tag{1.6}\]</span>
After calculating the Pearson coefficient of expression correlation, <span class="math inline">\(r_{12}=\frac{Cov\left(X_1,X_2\right)}{\sqrt{Var\left(X_1\right)Var\left(X_2\right)}}\)</span>, we obtain a simple formula for estimating the expression distance, that is,
<span class="math display">\[D_{12}=-\ln\left(r_{12}\right)=-\ln\left(1-P_{12}\right)\tag{1.7}\]</span>
When <span class="math inline">\(r_{12}\)</span> is close to 1 in the case of closely-related species, <span class="math inline">\(D_{12} \approx 1- r_{12}\)</span>, i.e., the Pearson distance <span class="math inline">\(P_{12}=1- r_{12}\)</span>.</li>
</ol>
</div>
<div id="expression-distance-estimation-when-expression-optima-vary-among-genes" class="section level3">
<h3><span class="header-section-number">4.1.5</span> Expression distance estimation when expression optima vary among genes</h3>
<p>While expression distance estimation based on Eq. (1.7) is intuitively simple, we have realized that the implied constant-<span class="math inline">\(\mu\)</span> assumption is unrealistic. Because the optimal expression level (<span class="math inline">\(\mu\)</span>) actually varies considerably among different genes, neglecting this variation may lead to an underestimation. Here we develop a new method to correct this bias.</p>
<p>Suppose that <span class="math inline">\(\mu\)</span> varies among genes according to a normal distribution with mean zero and variance <span class="math inline">\(V_{\mu}\)</span>. Under the stationary OU model, we can extend Eq.(1.4) as follows:</p>
<p>the expression variances are given by <span class="math inline">\(V_{11}=V_{22}=1/W+V_{\mu}\)</span> and the covariance by
<span class="math display">\[V_{12}=\frac{e^{-2\beta t}}{W}+V_{\mu}\tag{1.8}\]</span>
It follows that the Pearson coefficient of correlation <span class="math inline">\(r_{12}=\frac{V_{12}}{\sqrt{\left[V11,\  V22\right]}}\)</span> can be written as
<span class="math display">\[r_{12}=\frac{\left(e^{-2\beta t}+WV_{\mu}\right)}{\left(1+W_{\mu}\right)}=\pi+\left(1-\pi\right)^{e^{-2\beta t}}\tag{1.9}\]</span>
where <span class="math inline">\(\pi=\frac{WV_{\mu}}{\left(1+WV_{\mu}\right)}\)</span>, which measures the relative <span class="math inline">\(\pi=0\)</span> means among genes. While <span class="math inline">\(\pi=0\)</span> means a constant-<span class="math inline">\(\mu\)</span> among genes, <span class="math inline">\(\pi=1\)</span> means indicates a very strong <span class="math inline">\(\mu\)</span> variation among genes. <span class="math inline">\(r_{12}\rightarrow\pi\)</span> as <span class="math inline">\(t\rightarrow \infty\)</span>. From Eq.(9) we obtain a general formula to estimate the express distance <span class="math inline">\(D_{12}=2\beta t\)</span> as follows
<span class="math display">\[D_{12}=-\ln\frac{\left(r_{12}-\pi\right)}{\left(1-\pi\right)}=-\ln\left[1-\frac{P_{12}}{1-\pi}\right]\tag{1.10}\]</span>
where <span class="math inline">\(P_{12}=1-r_{12}\)</span> is the Pearson distance. Apparently, Eq.(1.10) is reduced to Eq.(1.7) when <span class="math inline">\(\pi=0\)</span>.</p>
</div>
<div id="estimation-of-pi" class="section level3">
<h3><span class="header-section-number">4.1.6</span> Estimation of <span class="math inline">\(\pi\)</span></h3>
<p>When Eq.(1.10) is applied in the evolutionary analysis of RNA-seq data, we have to know the parameter <span class="math inline">\(\pi\)</span>, which can be estimated when RNA-seq data of the same tissue from <span class="math inline">\(n&gt;=3\)</span> species are available.</p>
<p>Let <span class="math inline">\(x_k=\left(x_{1k},\cdots x_{nk}\right)\)</span> be the across-species expression profile of the <span class="math inline">\(k\)</span>-th orthologous gene; <span class="math inline">\(k=1,\cdots N\)</span>.</p>
<p>We first calculate the mean expression level of each <span class="math inline">\(k\)</span>-th orthologous gene, denoted by <span class="math inline">\(x_{.k}\)</span>, and then calculate the variance, <span class="math inline">\(Var\left(x_.\right)\)</span>, of <span class="math inline">\(x_{.k}\)</span> over all <span class="math inline">\(N\)</span> genes. It appears that <span class="math inline">\(Var\left(x_.\right)\)</span> is an asymptotically unbiased estimate of <span class="math inline">\(V_\mu\)</span>. Next, the strength of expression conservation (<span class="math inline">\(W\)</span>) can be estimated by a simple method in the following part.</p>
<p>It follows that <span class="math inline">\(\pi\)</span> can be estimated by</p>
<p><span class="math display">\[\pi=\frac{Var\left(x_{.}\right)W}{1+WVar(x_.)}\tag{1.11}\]</span>
</p>
</div>
</div>
<div id="case-study-expression-tree-building" class="section level2">
<h2><span class="header-section-number">4.2</span> Case Study: Expression Tree Building</h2>
<p>In here, we will give an example to build a character tree from expression data (expression phylogeny) applying the above-mentioned methods.</p>
<p><em>TreeExp</em> can be loaded in usual way:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(TreeExp)</code></pre>
<p>We first load the datasets created from six tissues’ expression data of nine tetrapod species</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;tetraExp&quot;</span>)</code></pre>
<div id="distance-matrix" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Distance matrix</h3>
<p>First, we generate an expression distance matrix of these nine tetrapod species:</p>
<pre class="sourceCode r"><code class="sourceCode r">dismat_pea &lt;-<span class="st"> </span><span class="kw">expdist</span>(tetraExp, <span class="dt">taxa =</span> <span class="st">&quot;all&quot;</span>,
                 <span class="dt">subtaxa =</span> <span class="st">&quot;Brain&quot;</span>,
                 <span class="dt">method =</span> <span class="st">&quot;pea&quot;</span>)
<span class="kw">as.dist</span>(dismat_pea)</code></pre>
<pre><code>##                  Human_Brain Chimpanzee_Brain Bonobo_Brain Gorilla_Brain
## Chimpanzee_Brain  0.03928696                                            
## Bonobo_Brain      0.05396840       0.02661862                           
## Gorilla_Brain     0.04542102       0.04346246   0.05571070              
## Orangutan_Brain   0.07440973       0.06649678   0.07417594    0.06650759
## Macaque_Brain     0.07097455       0.07025971   0.07569084    0.07142852
## Mouse_Brain       0.14753703       0.14734046   0.14569873    0.15774612
## Opossum_Brain     0.21459736       0.21023241   0.20623679    0.22219040
## Platypus_Brain    0.25207183       0.24840593   0.24225669    0.26264602
## Chicken_Brain     0.27146218       0.27435186   0.26609731    0.27169109
##                  Orangutan_Brain Macaque_Brain Mouse_Brain Opossum_Brain
## Chimpanzee_Brain                                                        
## Bonobo_Brain                                                            
## Gorilla_Brain                                                           
## Orangutan_Brain                                                         
## Macaque_Brain         0.07718357                                        
## Mouse_Brain           0.15908149    0.12541882                          
## Opossum_Brain         0.22389819    0.19697269  0.17580943              
## Platypus_Brain        0.26310593    0.23960340  0.22327614    0.20305065
## Chicken_Brain         0.26892835    0.24476930  0.23916935    0.28310733
##                  Platypus_Brain
## Chimpanzee_Brain               
## Bonobo_Brain                   
## Gorilla_Brain                  
## Orangutan_Brain                
## Macaque_Brain                  
## Mouse_Brain                    
## Opossum_Brain                  
## Platypus_Brain                 
## Chicken_Brain        0.28657956</code></pre>
<p><code>expdist</code> function can calculate expression distance directly through the <em>taxaExo</em> object. It will extract the expression values and <span class="math inline">\(log2\)</span>-transform the values according to the <strong>“taxa”</strong> and <strong>“subtaxa”</strong> specified in the options.</p>
<p>The default method <em>“pea”</em> is to calculate pair-wise distances by Pearson distance, which equals 1-Pearson’s coefficient of expression level.</p>
<p>Besides <em>“pea”</em> , there are numbers of alternative methods to calculate pair-wise expression distances, like<em>“sou”, “sou_v”,“pea”, “spe”,“euc”, “cos”, “jsd”, “tani”, “jac”</em>.</p>
<p>Of particular, <strong>“sou”</strong> means the general sOU distance method and <strong>“sou_v”</strong> means the special sOU model when expression optima vary among genes that we introduced above.</p>
<p>For comparison, we apply <strong>“pea”</strong>,<strong>“sou”</strong>and <strong>“sou_v”</strong> methods to calculate the paired expression distances between species.</p>
<p>Distance matrix applying <strong>“sou”</strong> method:</p>
<pre class="sourceCode r"><code class="sourceCode r">dismat_sou &lt;-<span class="st"> </span><span class="kw">expdist</span>(tetraExp, <span class="dt">taxa =</span> <span class="st">&quot;all&quot;</span>,
                 <span class="dt">subtaxa =</span> <span class="st">&quot;Brain&quot;</span>,
                 <span class="dt">method =</span> <span class="st">&quot;sou&quot;</span>)
<span class="kw">as.dist</span>(dismat_sou)</code></pre>
<pre><code>##                  Human_Brain Chimpanzee_Brain Bonobo_Brain Gorilla_Brain
## Chimpanzee_Brain  0.04007952                                            
## Bonobo_Brain      0.05547931       0.02697931                           
## Gorilla_Brain     0.04648489       0.04443524   0.05732270              
## Orangutan_Brain   0.07732362       0.06881087   0.07707107    0.06882245
## Macaque_Brain     0.07361915       0.07284999   0.07870868    0.07410792
## Mouse_Brain       0.15962550       0.15939495   0.15747137    0.17167379
## Opossum_Brain     0.24155877       0.23601657   0.23097009    0.25127352
## Platypus_Brain    0.29044833       0.28555890   0.27741060    0.30468720
## Chicken_Brain     0.31671573       0.32069003   0.30937883    0.31702999
##                  Orangutan_Brain Macaque_Brain Mouse_Brain Opossum_Brain
## Chimpanzee_Brain                                                        
## Bonobo_Brain                                                            
## Gorilla_Brain                                                           
## Orangutan_Brain                                                         
## Macaque_Brain         0.08032494                                        
## Mouse_Brain           0.17326052    0.13401016                          
## Opossum_Brain         0.25347156    0.21936656  0.19335350              
## Platypus_Brain        0.30531112    0.27391514  0.25267038    0.22696416
## Chicken_Brain         0.31324380    0.28073202  0.27334448    0.33282915
##                  Platypus_Brain
## Chimpanzee_Brain               
## Bonobo_Brain                   
## Gorilla_Brain                  
## Orangutan_Brain                
## Macaque_Brain                  
## Mouse_Brain                    
## Opossum_Brain                  
## Platypus_Brain                 
## Chicken_Brain        0.33768435</code></pre>
<p>Distance matrix applying <strong>“sou_v”</strong> method:</p>
<pre class="sourceCode r"><code class="sourceCode r">dismat_sou_v &lt;-<span class="st"> </span><span class="kw">expdist</span>(tetraExp, <span class="dt">taxa =</span> <span class="st">&quot;all&quot;</span>,
                 <span class="dt">subtaxa =</span> <span class="st">&quot;Brain&quot;</span>,
                 <span class="dt">method =</span> <span class="st">&quot;sou_v&quot;</span>)
dismat_sou_v<span class="op">$</span>pi</code></pre>
<pre><code>## [1] 0.4321824</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">as.dist</span>(dismat_sou_v<span class="op">$</span>distance)</code></pre>
<pre><code>##                  Human_Brain Chimpanzee_Brain Bonobo_Brain Gorilla_Brain
## Chimpanzee_Brain  0.07169946                                            
## Bonobo_Brain      0.09987040       0.04801322                           
## Gorilla_Brain     0.08337321       0.07963103   0.10326685              
## Orangutan_Brain   0.14046409       0.12455399   0.13999038    0.12457556
## Macaque_Brain     0.13352606       0.13208832   0.14306392    0.13444018
## Mouse_Brain       0.30087772       0.30041013   0.29651329    0.32546876
## Opossum_Brain     0.47470846       0.46242662   0.45131469    0.49643952
## Platypus_Brain    0.58686287       0.57531948   0.55625069    0.62092606
## Chicken_Brain     0.65024075       0.66003933   0.63229985    0.65101349
##                  Orangutan_Brain Macaque_Brain Mouse_Brain Opossum_Brain
## Chimpanzee_Brain                                                        
## Bonobo_Brain                                                            
## Gorilla_Brain                                                           
## Orangutan_Brain                                                         
## Macaque_Brain         0.14610174                                        
## Mouse_Brain           0.32873050    0.24958856                          
## Opossum_Brain         0.50139288    0.42601631  0.37051757              
## Platypus_Brain        0.62243425    0.54813379  0.49958580    0.44254159
## Chicken_Brain         0.64172713    0.56399841  0.54681221    0.69032820
##                  Platypus_Brain
## Chimpanzee_Brain               
## Bonobo_Brain                   
## Gorilla_Brain                  
## Orangutan_Brain                
## Macaque_Brain                  
## Mouse_Brain                    
## Opossum_Brain                  
## Platypus_Brain                 
## Chicken_Brain        0.70259882</code></pre>
<p>Note that method <strong>“sou_v”</strong> will return an object of <em>list</em> which contains two element. The firt one is the estimated value of parameter <span class="math inline">\(\pi\)</span>. The second one is the distance matrix using <strong>“sou_v”</strong> method.</p>
<pre class="sourceCode r"><code class="sourceCode r">dismat_sou_v<span class="op">$</span>pi</code></pre>
<pre><code>## [1] 0.4321824</code></pre>
<p>Comparing the distance values calculted by the three methods, we can approximately draw a conclusion that while the optimal expression level(<span class="math inline">\(\mu\)</span>) actually varies considerably among different genes, neglecting this variation may lead to an underesitmation of expression distance.</p>
<p>Also, if you already have a data frame with normalized expression values, there are internal functions available for creating expression distance matrix.</p>
<p>Function <code>exptabTE</code> is helpful to generate an expression level table from the <em>taxaExp</em> object and <span class="math inline">\(log2\)</span>-transform the values. Also, <strong>“taxa”</strong>&quot; and <strong>“subtaxa”</strong>&quot; can be set by users.
For instance,</p>
<pre class="sourceCode r"><code class="sourceCode r">expression_table &lt;-<span class="st"> </span><span class="kw">exptabTE</span>(tetraExp, <span class="dt">taxa =</span> <span class="st">&quot;all&quot;</span>,
                            <span class="dt">subtaxa =</span> <span class="st">&quot;Brain&quot;</span>)

dismat &lt;-<span class="st"> </span><span class="kw">dist.pea</span>(expression_table)
dismat &lt;-<span class="st"> </span><span class="kw">dist.sou</span>(expression_table)
dismat &lt;-<span class="st"> </span><span class="kw">dist.sou_v</span>(expression_table)</code></pre>
<p>If you have your own expression data frame in the format as the “expression_table” here, it will do fine:</p>
<pre class="sourceCode r"><code class="sourceCode r">dismat &lt;-<span class="st"> </span><span class="kw">dist.pea</span>(your_own_dataframe)
<span class="kw">colnames</span>(dismat) &lt;-<span class="st"> </span><span class="kw">colnames</span>(your_own_dataframe)
<span class="kw">rownames</span>(dismat) &lt;-<span class="st"> </span><span class="kw">colnames</span>(dismat)</code></pre>

</div>
<div id="expression-character-tree" class="section level3">
<h3><span class="header-section-number">4.2.2</span> Expression character tree</h3>
<p>After the expression distance matrix is created, we can construct character tree by Neighbor-Joining method, and bootstrap values based on re-sampling orthologous genes with replacements can also be generated by <code>boot.exphy</code> function:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&#39;ape&#39;</span>)
 ##### build NJ tree
tr &lt;-<span class="st"> </span><span class="kw">NJ</span>(dismat_sou)
 ### estract expression table of Brain tissue from &#39;tetraExp&#39;
expression_table &lt;-<span class="kw">exptabTE</span>(<span class="dt">objects =</span> tetraExp,<span class="dt">taxa =</span> <span class="st">&#39;all&#39;</span>,
                            <span class="dt">subtaxa =</span> <span class="st">&#39;Brain&#39;</span>)
### root the tree using &#39;Chicken_Brain&#39; as outgroup
tr &lt;-<span class="st"> </span><span class="kw">root</span>(tr, <span class="st">&quot;Chicken_Brain&quot;</span>, <span class="dt">resolve.root =</span> <span class="ot">TRUE</span>) 
###### generating bootstrap values of the tree
bs&lt;-<span class="kw">boot.exphy</span>(<span class="dt">phy=</span>tr, <span class="dt">x =</span> expression_table, <span class="dt">method =</span> <span class="st">&#39;sou&#39;</span>,
               <span class="dt">outgroup =</span> <span class="st">&#39;Chicken_Brain&#39;</span>, <span class="dt">B =</span> <span class="dv">100</span>) 
### assign bootstrap values to the &quot;node.label&quot; of the tree
tr<span class="op">$</span>node.label =<span class="st"> </span>bs 
### plot the tree
<span class="kw">plot</span>(tr, <span class="dt">show.node.label =</span> <span class="ot">TRUE</span>)</code></pre>
<p><img src="TreeExp-Tutorial_files/figure-html/unnamed-chunk-23-1.png" width="576" /></p>
<p>Note that the goal of <code>root</code> step is to give an better interpretation on the phylogentic relationships of trascriptome data.</p>
<p>Essentially, the tree build by NJ method itself is an unrooted tree. Though <code>root</code> function adds a ‘root’ node to the tree, the branch length between root node and its neighbor internal node is still 0. In other words, <code>root</code> step may be only for tree presentation, and it is not needed when a phylogentic tree should be used as input parameter in the following analysis (such like <em>Ancestral Transcriptome Inference</em>).</p>
<p>By now, an expression character tree is successfully constructed. The tree shows expression patterns’ similarities in selected genes of designated species. The expression tree is largely in accordance with species tree with minor discrepancy.</p>
<p>Phenomenon of evolutionary history dominates the evolutionary expression pattern can be described as phylogenetic signals. One way to interpret highly consistent expression character tree is that expression levels of transcriptome, representing the regulatory changes, accumulated over time. Though not as concrete as sequence data, expression levels generated from transcriptome data across species show strong phylogenetic signals.</p>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-bergmann2004">
<p>Bergmann, Sven, Jan Ihmels, and Naama Barkai. 2004. “Similarities and Differences in Genome-Wide Expression Data of Six Organisms.” <em>PLoS Biology</em> 2:e9. <a href="https://doi.org/10.1371/journal.pbio.0020009" class="uri">https://doi.org/10.1371/journal.pbio.0020009</a>.</p>
</div>
<div id="ref-butler2004">
<p>Butler, and King. 2004. “Phylogenetic Comparative Analysis: A Modeling Approach for Adaptive Evolution.” <em>The American Naturalist</em> 164:683. <a href="https://doi.org/10.2307/3473229" class="uri">https://doi.org/10.2307/3473229</a>.</p>
</div>
<div id="ref-hansen1996">
<p>Hansen, Thomas F, and Emilia P Martins. 1996. “Translating Between Microevolutionary Process and Macroevolutionary Patterns: The Correlation Structure of Interspecific Data.” <em>Evolution</em> 50:1404. <a href="https://doi.org/10.2307/2410878" class="uri">https://doi.org/10.2307/2410878</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-manipulation-and-storage.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="relative-rate-test-for-transcriptome-evolution.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03-Distance.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["TreeExp-Tutorial.pdf", "TreeExp-Tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
