<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.10 and GitBook 2.6.7" />

  <meta property="og:title" content="References | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Jingwen Yang" />


<meta name="date" content="2019-05-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="ancestral-transcriptome-inference.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"><i>TreeExp</i>-Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#scope"><i class="fa fa-check"></i><b>2.1</b> Scope</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#citation"><i class="fa fa-check"></i><b>2.3</b> Citation</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#how-to-get-help"><i class="fa fa-check"></i><b>2.4</b> How to Get Help?</a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#rna-seq-data-enbeded"><i class="fa fa-check"></i><b>2.5</b> RNA-seq Data Enbeded</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html"><i class="fa fa-check"></i><b>3</b> Data Manipulation and Storage</a><ul>
<li class="chapter" data-level="3.1" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#input-format"><i class="fa fa-check"></i><b>3.1</b> Input Format</a></li>
<li class="chapter" data-level="3.2" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#example-of-input-data"><i class="fa fa-check"></i><b>3.2</b> Example of Input Data</a></li>
<li class="chapter" data-level="3.3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#construction"><i class="fa fa-check"></i><b>3.3</b> Construction</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html"><i class="fa fa-check"></i><b>4</b> Pairwise Expression Distance and Expression Tree-Making</a><ul>
<li class="chapter" data-level="4.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#theory"><i class="fa fa-check"></i><b>4.1</b> Theory</a><ul>
<li class="chapter" data-level="4.1.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-ornstein-uhlenbeck-ou-model-of-transcriptome-evolution"><i class="fa fa-check"></i><b>4.1.1</b> The Ornstein-Uhlenbeck (OU) model of transcriptome evolution</a></li>
<li class="chapter" data-level="4.1.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#transcriptome-evolution-between-species"><i class="fa fa-check"></i><b>4.1.2</b> Transcriptome evolution between species</a></li>
<li class="chapter" data-level="4.1.3" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-estimation-problem-of-expression-distance"><i class="fa fa-check"></i><b>4.1.3</b> The estimation problem of expression distance</a></li>
<li class="chapter" data-level="4.1.4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-expression-distance-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>4.1.4</b> Estimation of expression distance under the stationary OU model</a></li>
<li class="chapter" data-level="4.1.5" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-distance-estimation-when-expression-optima-vary-among-genes"><i class="fa fa-check"></i><b>4.1.5</b> Expression distance estimation when expression optima vary among genes</a></li>
<li class="chapter" data-level="4.1.6" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-pi"><i class="fa fa-check"></i><b>4.1.6</b> Estimation of <span class="math inline">\(\pi\)</span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#case-study-expression-tree-building"><i class="fa fa-check"></i><b>4.2</b> Case Study: Expression Tree Building</a><ul>
<li class="chapter" data-level="4.2.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#distance-matrix"><i class="fa fa-check"></i><b>4.2.1</b> Distance matrix</a></li>
<li class="chapter" data-level="4.2.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-character-tree"><i class="fa fa-check"></i><b>4.2.2</b> Expression character tree</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html"><i class="fa fa-check"></i><b>5</b> Relative Rate Test for Transcriptome Evolution</a><ul>
<li class="chapter" data-level="5.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#theory-1"><i class="fa fa-check"></i><b>5.1</b> Theory</a></li>
<li class="chapter" data-level="5.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#statistical-procedure-of-the-relative-rate-test"><i class="fa fa-check"></i><b>5.2</b> Statistical Procedure of the Relative Rate Test</a><ul>
<li class="chapter" data-level="5.2.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#sampling-variance-of-the-expression-distance"><i class="fa fa-check"></i><b>5.2.1</b> Sampling variance of the expression distance</a></li>
<li class="chapter" data-level="5.2.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#calculation-of-varbigtriangleup_ab"><i class="fa fa-check"></i><b>5.2.2</b> Calculation of <span class="math inline">\(Var(\bigtriangleup_{AB})\)</span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#case-study-testing-fast-evolving-genes-in-human"><i class="fa fa-check"></i><b>5.3</b> Case Study: Testing Fast-evolving Genes in Human</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html"><i class="fa fa-check"></i><b>6</b> Estimating the Strength of Expression Conservation</a><ul>
<li class="chapter" data-level="6.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#theory-2"><i class="fa fa-check"></i><b>6.1</b> Theory</a><ul>
<li class="chapter" data-level="6.1.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stabilizing-selection-model-in-transcriptome-evolution-across-species"><i class="fa fa-check"></i><b>6.1.1</b> Stabilizing selection model in transcriptome evolution across species</a></li>
<li class="chapter" data-level="6.1.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stationary-ou-model-under-a-species-phylogeny"><i class="fa fa-check"></i><b>6.1.2</b> Stationary OU model under a species phylogeny</a></li>
<li class="chapter" data-level="6.1.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#variation-of-w-among-genes"><i class="fa fa-check"></i><b>6.1.3</b> Variation of <span class="math inline">\(W\)</span> among genes</a></li>
<li class="chapter" data-level="6.1.4" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#an-empirical-bayesian-framework-for-gene-specific-w-esitimation"><i class="fa fa-check"></i><b>6.1.4</b> An empirical Bayesian framework for gene-specific <span class="math inline">\(W\)</span> esitimation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#statistical-procedure"><i class="fa fa-check"></i><b>6.2</b> Statistical Procedure</a></li>
<li class="chapter" data-level="6.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#case-study-emprical-bayesian-esitimates-of-expression-conservation-of-genes"><i class="fa fa-check"></i><b>6.3</b> Case Study: Emprical Bayesian Esitimates of Expression Conservation of genes</a><ul>
<li class="chapter" data-level="6.3.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#inversed-correlation-matrix"><i class="fa fa-check"></i><b>6.3.1</b> Inversed correlation matrix</a></li>
<li class="chapter" data-level="6.3.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#estimation-of-gamma-parameters"><i class="fa fa-check"></i><b>6.3.2</b> Estimation of gamma parameters</a></li>
<li class="chapter" data-level="6.3.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#bayesian-estimation-of-gene-specific-selection-pressure"><i class="fa fa-check"></i><b>6.3.3</b> Bayesian estimation of gene-specific selection pressure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html"><i class="fa fa-check"></i><b>7</b> Ancestral Transcriptome Inference</a><ul>
<li class="chapter" data-level="7.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#theory-3"><i class="fa fa-check"></i><b>7.1</b> Theory</a><ul>
<li class="chapter" data-level="7.1.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#from-brownian-motion-bm-model-to-ornstein-uhlenbeck-ou-model"><i class="fa fa-check"></i><b>7.1.1</b> From Brownian Motion (BM) model to Ornstein-Uhlenbeck (OU) model</a></li>
<li class="chapter" data-level="7.1.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-ou-model-under-a-phylogeny"><i class="fa fa-check"></i><b>7.1.2</b> The OU model under a phylogeny</a></li>
<li class="chapter" data-level="7.1.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-gene-expression-inference-an-empirical-bayesian-approach"><i class="fa fa-check"></i><b>7.1.3</b> Ancestral gene expression inference: an empirical Bayesian approach</a></li>
<li class="chapter" data-level="7.1.4" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-problem-of-ancestral-transcriptome-inference"><i class="fa fa-check"></i><b>7.1.4</b> The problem of ancestral transcriptome inference</a></li>
<li class="chapter" data-level="7.1.5" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#stationary-ornstein-uhlenbeck-sou-model"><i class="fa fa-check"></i><b>7.1.5</b> Stationary Ornstein-Uhlenbeck (sOU) model</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#a-fast-algorithm-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>7.2</b> A Fast Algorithm under the Stationary OU Model</a></li>
<li class="chapter" data-level="7.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#case-study-ancestral-expression-inference-of-gene-emp1"><i class="fa fa-check"></i><b>7.3</b> Case Study: Ancestral Expression Inference of Gene <em>EMP1</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#expression-character-tree-1"><i class="fa fa-check"></i><b>7.3.1</b> Expression character tree</a></li>
<li class="chapter" data-level="7.3.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#creating-variance-co-variance-matrix"><i class="fa fa-check"></i><b>7.3.2</b> Creating variance co-variance matrix</a></li>
<li class="chapter" data-level="7.3.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-expression-estimation"><i class="fa fa-check"></i><b>7.3.3</b> Ancestral expression estimation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/jingwyang/TreeExp" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial: Use <em>TreeExp</em> for Phylogenetic Transcriptome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references">
<div>
<p>Bedford, Trevor, and Daniel L. Hartl. 2009. “Optimization of gene expression by natural selection.” <em>Proceedings of the National Academy of Sciences</em> 106:1133–8. <a href="https://doi.org/10.1073/pnas.0812009106" class="uri">https://doi.org/10.1073/pnas.0812009106</a>.</p>
</div>
<div>
<p>Bergmann, Sven, Jan Ihmels, and Naama Barkai. 2004. “Similarities and Differences in Genome-Wide Expression Data of Six Organisms.” <em>PLoS Biology</em> 2:e9. <a href="https://doi.org/10.1371/journal.pbio.0020009" class="uri">https://doi.org/10.1371/journal.pbio.0020009</a>.</p>
</div>
<div>
<p>Brawand, David, Magali Soumillon, Anamaria Necsulea, Philippe Julien, Gábor Csárdi, Patrick Harrigan, Manuela Weier, et al. 2011. “The evolution of gene expression levels in mammalian organs.” <em>Nature</em> 478:343. <a href="https://doi.org/10.1038/nature10532" class="uri">https://doi.org/10.1038/nature10532</a>.</p>
</div>
<div>
<p>Butler, and King. 2004. “Phylogenetic Comparative Analysis: A Modeling Approach for Adaptive Evolution.” <em>The American Naturalist</em> 164:683. <a href="https://doi.org/10.2307/3473229" class="uri">https://doi.org/10.2307/3473229</a>.</p>
</div>
<div>
<p>Cui, Qinghua, Zhenbao Yu, Enrico O. Purisima, and Edwin Wang. 2007. “MicroRNA regulation and interspecific variation of gene expression.” <em>Trends in Genetics</em> 23:372–75. <a href="https://doi.org/10.1016/j.tig.2007.04.003" class="uri">https://doi.org/10.1016/j.tig.2007.04.003</a>.</p>
</div>
<div>
<p>Gu, Xun. 2004. “Statistical Framework for Phylogenomic Analysis of Gene Family Expression Profiles.” <em>Genetics</em> 167:531–42. <a href="https://doi.org/10.1534/genetics.167.1.531" class="uri">https://doi.org/10.1534/genetics.167.1.531</a>.</p>
</div>
<div>
<p>Gu, Xun, and Zhixi Su. 2007. “Tissue-driven hypothesis of genomic evolution and sequence-expression correlations.” <em>Proceedings of the National Academy of Sciences</em> 104:2779–84. <a href="https://doi.org/10.1073/pnas.0610797104" class="uri">https://doi.org/10.1073/pnas.0610797104</a>.</p>
</div>
<div>
<p>Gu, Xun, Hang Ruan, and Jingwen Yang. 2019. “Estimating the Strength of Expression Conservation from High Throughput RNA-seq Data.” <em>Bioinformatics</em>, May. <a href="https://doi.org/10.1093/bioinformatics/btz405" class="uri">https://doi.org/10.1093/bioinformatics/btz405</a>.</p>
</div>
<div>
<p>Hansen, Thomas F, and Emilia P Martins. 1996. “Translating Between Microevolutionary Process and Macroevolutionary Patterns: The Correlation Structure of Interspecific Data.” <em>Evolution</em> 50:1404. <a href="https://doi.org/10.2307/2410878" class="uri">https://doi.org/10.2307/2410878</a>.</p>
</div>
<div>
<p>Harrison, Peter W., Alison E. Wright, and Judith E. Mank. 2012. “The evolution of gene expression and the transcriptome–phenotype relationship.” <em>Seminars in Cell &amp; Developmental Biology</em> 23:222–29. <a href="https://doi.org/10.1016/j.semcdb.2011.12.004" class="uri">https://doi.org/10.1016/j.semcdb.2011.12.004</a>.</p>
</div>
<div>
<p>King, MC, and AC Wilson. 1975. “Evolution at two levels in humans and chimpanzees.” <em>Science</em> 188:107–16. <a href="https://doi.org/10.1126/science.1090005" class="uri">https://doi.org/10.1126/science.1090005</a>.</p>
</div>
<div>
<p>Lehner, Ben. 2013. “Genotype to phenotype: lessons from model organisms for human genetics.” <em>Nature Reviews Genetics</em> 14:168. <a href="https://doi.org/10.1038/nrg3404" class="uri">https://doi.org/10.1038/nrg3404</a>.</p>
</div>
<div>
<p>Lemos, Bernardo, Colin D. Meiklejohn, Mario Cáceres, and Daniel L. Hartl. 2005. “Rate of divergence in gene expression profiles of primates, mice and flies: stabilizing selection and variability among functionnal categories.” <em>Evolution</em> 63:126. <a href="https://doi.org/10.1554/04-251" class="uri">https://doi.org/10.1554/04-251</a>.</p>
</div>
<div>
<p>Park, Solip, and Ben Lehner. 2013. “Epigenetic epistatic interactions constrain the evolution of gene expression.” <em>Molecular Systems Biology</em> 9:645. <a href="https://doi.org/10.1038/msb.2013.2" class="uri">https://doi.org/10.1038/msb.2013.2</a>.</p>
</div>
<div>
<p>Robinson, Mark D., Davis J. McCarthy, and Gordon K. Smyth. 2010. “edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.” <em>Bioinformatics</em> 26:139–40. <a href="https://doi.org/10.1093/bioinformatics/btp616" class="uri">https://doi.org/10.1093/bioinformatics/btp616</a>.</p>
</div>
<div>
<p>Rohlfs, Rori V., Patrick Harrigan, and Rasmus Nielsen. 2014. “Modeling Gene Expression Evolution with an Extended Ornstein–Uhlenbeck Process Accounting for Within-Species Variation.” <em>Molecular Biology and Evolution</em> 31:201–11. <a href="https://doi.org/10.1093/molbev/mst190" class="uri">https://doi.org/10.1093/molbev/mst190</a>.</p>
</div>
<div>
<p>Ruan, Hang, Zhixi Su, and Xun Gu. 2016. “TreeExp1.0: R Package for Analyzing Expression Evolution Based on Rna-Seq Data.” <em>Journal of Experimental Zoology Part B: Molecular and Developmental Evolution</em> 326 (7):394–402. <a href="https://doi.org/10.1002/jez.b.22707" class="uri">https://doi.org/10.1002/jez.b.22707</a>.</p>
</div>
<div>
<p>Tirosh, Itay, Adina Weinberger, Miri Carmi, and Naama Barkai. 2006. “A genetic signature of interspecies variations in gene expression.” <em>Nature Genetics</em> 38:ng1819. <a href="https://doi.org/10.1038/ng1819" class="uri">https://doi.org/10.1038/ng1819</a>.</p>
</div>
<div>
<p>Warnefors, Maria, and Adam Eyre-Walker. 2012. “A Selection Index for Gene Expression Evolution and Its Application to the Divergence between Humans and Chimpanzees.” <em>PLoS ONE</em> 7:e34935. <a href="https://doi.org/10.1371/journal.pone.0034935" class="uri">https://doi.org/10.1371/journal.pone.0034935</a>.</p>
</div>
<div>
<p>Xu, Chuan, Qian Li, Olga Efimova, Liu He, Shoji Tatsumoto, Vita Stepanova, Takao Oishi, et al. 2018. “Human-specific features of spatial gene expression and regulation in eight brain regions.” <em>Genome Research</em> 28:1097–1110. <a href="https://doi.org/10.1101/gr.231357.117" class="uri">https://doi.org/10.1101/gr.231357.117</a>.</p>
</div>
<div>
<p>Yang, Jingwen, Hang Ruan, Yangyun Zou, Zhixi Su, and Xun Gu. 2018. “Ancestral Transcriptome Inference Based on Rna-Seq and Chip-Seq Data.” <em>Methods</em>. <a href="https://doi.org/10.1016/j.ymeth.2018.11.010" class="uri">https://doi.org/10.1016/j.ymeth.2018.11.010</a>.</p>
</div>
<div>
<p>Zou, Yangyun, Wei Huang, Zhenglong Gu, and Xun Gu. 2011. “Predominant Gain of Promoter TATA Box after Gene Duplication Associated with Stress Responses.” <em>Molecular Biology and Evolution</em> 28:2893–2904. <a href="https://doi.org/10.1093/molbev/msr116" class="uri">https://doi.org/10.1093/molbev/msr116</a>.</p>
</div>
</div>
</div>







            </section>

          </div>
        </div>
      </div>
<a href="ancestral-transcriptome-inference.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/07-Reference.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["TreeExp-Tutorial.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
