<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Relative Rate Test for Transcriptome Evolution | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis</title>
  <meta name="description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.10 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Relative Rate Test for Transcriptome Evolution | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Relative Rate Test for Transcriptome Evolution | Tutorial: Use TreeExp for Phylogenetic Transcriptome Analysis" />
  
  <meta name="twitter:description" content="This is a minimal example of using the bookdown package to write a book. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Jingwen Yang" />


<meta name="date" content="2019-05-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="pairwise-expression-distance-and-expression-tree-making.html">
<link rel="next" href="estimating-the-strength-of-expression-conservation.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"><i>TreeExp</i>-Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#scope"><i class="fa fa-check"></i><b>2.1</b> Scope</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#installation"><i class="fa fa-check"></i><b>2.2</b> Installation</a></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#citation"><i class="fa fa-check"></i><b>2.3</b> Citation</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#how-to-get-help"><i class="fa fa-check"></i><b>2.4</b> How to Get Help?</a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#rna-seq-data-enbeded"><i class="fa fa-check"></i><b>2.5</b> RNA-seq Data Enbeded</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html"><i class="fa fa-check"></i><b>3</b> Data Manipulation and Storage</a><ul>
<li class="chapter" data-level="3.1" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#input-format"><i class="fa fa-check"></i><b>3.1</b> Input Format</a></li>
<li class="chapter" data-level="3.2" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#example-of-input-data"><i class="fa fa-check"></i><b>3.2</b> Example of Input Data</a></li>
<li class="chapter" data-level="3.3" data-path="data-manipulation-and-storage.html"><a href="data-manipulation-and-storage.html#construction"><i class="fa fa-check"></i><b>3.3</b> Construction</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html"><i class="fa fa-check"></i><b>4</b> Pairwise Expression Distance and Expression Tree-Making</a><ul>
<li class="chapter" data-level="4.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#theory"><i class="fa fa-check"></i><b>4.1</b> Theory</a><ul>
<li class="chapter" data-level="4.1.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-ornstein-uhlenbeck-ou-model-of-transcriptome-evolution"><i class="fa fa-check"></i><b>4.1.1</b> The Ornstein-Uhlenbeck (OU) model of transcriptome evolution</a></li>
<li class="chapter" data-level="4.1.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#transcriptome-evolution-between-species"><i class="fa fa-check"></i><b>4.1.2</b> Transcriptome evolution between species</a></li>
<li class="chapter" data-level="4.1.3" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#the-estimation-problem-of-expression-distance"><i class="fa fa-check"></i><b>4.1.3</b> The estimation problem of expression distance</a></li>
<li class="chapter" data-level="4.1.4" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-expression-distance-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>4.1.4</b> Estimation of expression distance under the stationary OU model</a></li>
<li class="chapter" data-level="4.1.5" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-distance-estimation-when-expression-optima-vary-among-genes"><i class="fa fa-check"></i><b>4.1.5</b> Expression distance estimation when expression optima vary among genes</a></li>
<li class="chapter" data-level="4.1.6" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#estimation-of-pi"><i class="fa fa-check"></i><b>4.1.6</b> Estimation of <span class="math inline">\(\pi\)</span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#case-study-expression-tree-building"><i class="fa fa-check"></i><b>4.2</b> Case Study: Expression Tree Building</a><ul>
<li class="chapter" data-level="4.2.1" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#distance-matrix"><i class="fa fa-check"></i><b>4.2.1</b> Distance matrix</a></li>
<li class="chapter" data-level="4.2.2" data-path="pairwise-expression-distance-and-expression-tree-making.html"><a href="pairwise-expression-distance-and-expression-tree-making.html#expression-character-tree"><i class="fa fa-check"></i><b>4.2.2</b> Expression character tree</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html"><i class="fa fa-check"></i><b>5</b> Relative Rate Test for Transcriptome Evolution</a><ul>
<li class="chapter" data-level="5.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#theory-1"><i class="fa fa-check"></i><b>5.1</b> Theory</a></li>
<li class="chapter" data-level="5.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#statistical-procedure-of-the-relative-rate-test"><i class="fa fa-check"></i><b>5.2</b> Statistical Procedure of the Relative Rate Test</a><ul>
<li class="chapter" data-level="5.2.1" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#sampling-variance-of-the-expression-distance"><i class="fa fa-check"></i><b>5.2.1</b> Sampling variance of the expression distance</a></li>
<li class="chapter" data-level="5.2.2" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#calculation-of-varbigtriangleup_ab"><i class="fa fa-check"></i><b>5.2.2</b> Calculation of <span class="math inline">\(Var(\bigtriangleup_{AB})\)</span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="relative-rate-test-for-transcriptome-evolution.html"><a href="relative-rate-test-for-transcriptome-evolution.html#case-study-testing-fast-evolving-genes-in-human"><i class="fa fa-check"></i><b>5.3</b> Case Study: Testing Fast-evolving Genes in Human</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html"><i class="fa fa-check"></i><b>6</b> Estimating the Strength of Expression Conservation</a><ul>
<li class="chapter" data-level="6.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#theory-2"><i class="fa fa-check"></i><b>6.1</b> Theory</a><ul>
<li class="chapter" data-level="6.1.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stabilizing-selection-model-in-transcriptome-evolution-across-species"><i class="fa fa-check"></i><b>6.1.1</b> Stabilizing selection model in transcriptome evolution across species</a></li>
<li class="chapter" data-level="6.1.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#stationary-ou-model-under-a-species-phylogeny"><i class="fa fa-check"></i><b>6.1.2</b> Stationary OU model under a species phylogeny</a></li>
<li class="chapter" data-level="6.1.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#variation-of-w-among-genes"><i class="fa fa-check"></i><b>6.1.3</b> Variation of <span class="math inline">\(W\)</span> among genes</a></li>
<li class="chapter" data-level="6.1.4" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#an-empirical-bayesian-framework-for-gene-specific-w-esitimation"><i class="fa fa-check"></i><b>6.1.4</b> An empirical Bayesian framework for gene-specific <span class="math inline">\(W\)</span> esitimation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#statistical-procedure"><i class="fa fa-check"></i><b>6.2</b> Statistical Procedure</a></li>
<li class="chapter" data-level="6.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#case-study-emprical-bayesian-esitimates-of-expression-conservation-of-genes"><i class="fa fa-check"></i><b>6.3</b> Case Study: Emprical Bayesian Esitimates of Expression Conservation of genes</a><ul>
<li class="chapter" data-level="6.3.1" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#inversed-correlation-matrix"><i class="fa fa-check"></i><b>6.3.1</b> Inversed correlation matrix</a></li>
<li class="chapter" data-level="6.3.2" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#estimation-of-gamma-parameters"><i class="fa fa-check"></i><b>6.3.2</b> Estimation of gamma parameters</a></li>
<li class="chapter" data-level="6.3.3" data-path="estimating-the-strength-of-expression-conservation.html"><a href="estimating-the-strength-of-expression-conservation.html#bayesian-estimation-of-gene-specific-selection-pressure"><i class="fa fa-check"></i><b>6.3.3</b> Bayesian estimation of gene-specific selection pressure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html"><i class="fa fa-check"></i><b>7</b> Ancestral Transcriptome Inference</a><ul>
<li class="chapter" data-level="7.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#theory-3"><i class="fa fa-check"></i><b>7.1</b> Theory</a><ul>
<li class="chapter" data-level="7.1.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#from-brownian-motion-bm-model-to-ornstein-uhlenbeck-ou-model"><i class="fa fa-check"></i><b>7.1.1</b> From Brownian Motion (BM) model to Ornstein-Uhlenbeck (OU) model</a></li>
<li class="chapter" data-level="7.1.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-ou-model-under-a-phylogeny"><i class="fa fa-check"></i><b>7.1.2</b> The OU model under a phylogeny</a></li>
<li class="chapter" data-level="7.1.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-gene-expression-inference-an-empirical-bayesian-approach"><i class="fa fa-check"></i><b>7.1.3</b> Ancestral gene expression inference: an empirical Bayesian approach</a></li>
<li class="chapter" data-level="7.1.4" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#the-problem-of-ancestral-transcriptome-inference"><i class="fa fa-check"></i><b>7.1.4</b> The problem of ancestral transcriptome inference</a></li>
<li class="chapter" data-level="7.1.5" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#stationary-ornstein-uhlenbeck-sou-model"><i class="fa fa-check"></i><b>7.1.5</b> Stationary Ornstein-Uhlenbeck (sOU) model</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#a-fast-algorithm-under-the-stationary-ou-model"><i class="fa fa-check"></i><b>7.2</b> A Fast Algorithm under the Stationary OU Model</a></li>
<li class="chapter" data-level="7.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#case-study-ancestral-expression-inference-of-gene-emp1"><i class="fa fa-check"></i><b>7.3</b> Case Study: Ancestral Expression Inference of Gene <em>EMP1</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#expression-character-tree-1"><i class="fa fa-check"></i><b>7.3.1</b> Expression character tree</a></li>
<li class="chapter" data-level="7.3.2" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#creating-variance-co-variance-matrix"><i class="fa fa-check"></i><b>7.3.2</b> Creating variance co-variance matrix</a></li>
<li class="chapter" data-level="7.3.3" data-path="ancestral-transcriptome-inference.html"><a href="ancestral-transcriptome-inference.html#ancestral-expression-estimation"><i class="fa fa-check"></i><b>7.3.3</b> Ancestral expression estimation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/jingwyang/TreeExp" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Tutorial: Use <em>TreeExp</em> for Phylogenetic Transcriptome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="relative-rate-test-for-transcriptome-evolution" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Relative Rate Test for Transcriptome Evolution</h1>
<div id="theory-1" class="section level2">
<h2><span class="header-section-number">5.1</span> Theory</h2>
<p>Given a set (<span class="math inline">\(N\)</span>) of orthologous genes of two species denoted by <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span>, respectively, the goal of relative rate test is to investigate whether the rate of transcriptome evolution, on average, differs significantly between two lineages of species <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span>. To this end, we need the third species (species <span class="math inline">\(C\)</span>) as outgroup (<strong>Figure 1B</strong>).</p>
<p>Let <span class="math inline">\(D_{AB}\)</span>, <span class="math inline">\(D_{AC}\)</span> and <span class="math inline">\(D_{BC}\)</span> be the pairwise expression distances, respectively, which can be estimated by the methods we formulated above. Let <span class="math inline">\(a\)</span>, <span class="math inline">\(b\)</span>, <span class="math inline">\(c\)</span> be the expression branch lengths of corresponding three lineages <span class="math inline">\(A\)</span>, <span class="math inline">\(B\)</span> and <span class="math inline">\(C\)</span>, respectively. Assuming that these expression distances are additive, we have</p>
<p><span class="math display">\[
\begin{split}
D_{AB}&amp;=a+b \\
D_{AC}&amp;=a+c \\
D_{BC}&amp;=b+c
\end{split}\tag{2.1}
\]</span></p>
<p>respectively. Since lineages <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span> have the same evolutionary time (<span class="math inline">\(t\)</span>), one may write <span class="math inline">\(a=\beta_At\)</span> and <span class="math inline">\(b=\beta_Bt\)</span>, where <span class="math inline">\(\beta_A\)</span> and <span class="math inline">\(\beta_B\)</span> are the rates of transcriptome evolution in lineages <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span>, respectively.</p>
<p>The relative rate test considers the following statistic
<span class="math display">\[G_{AB}=D_{AC}-D_{BC}=a-b=\left(\beta_A-\beta_B\right)t\tag{2.2}\]</span></p>
<p>Hence, the null hypothesis <span class="math inline">\(G_{AB}=0\)</span> or <span class="math inline">\(D_{AC}=D_{BC}\)</span> means an equal rate (<span class="math inline">\(\beta_A=\beta_B\)</span>) of expression divergence between two lineages. Rejection of this null indicates a rapid expression evolution in lineage <span class="math inline">\(A\)</span> (<span class="math inline">\(\beta_A&gt;\beta_B\)</span> if <span class="math inline">\(G_{AB}&gt;0\)</span>) or in lineage <span class="math inline">\(B\)</span> (<span class="math inline">\(\beta_A&gt;\beta_B\)</span> if <span class="math inline">\(G_{AB}&lt;0\)</span>).</p>
<p>Here, we apply a Z-score test to examine the signigicance</p>
<p><span class="math display">\[Z\  =\  \frac{\bigtriangleup_{AB}}{\sqrt{Var\left(\bigtriangleup_{AB}\right)}}\tag{2.3}\]</span></p>
<p>where <span class="math inline">\(\bigtriangleup_{AB}\)</span> equals to <span class="math inline">\(G_{AB}\)</span>.</p>

</div>
<div id="statistical-procedure-of-the-relative-rate-test" class="section level2">
<h2><span class="header-section-number">5.2</span> Statistical Procedure of the Relative Rate Test</h2>
<div id="sampling-variance-of-the-expression-distance" class="section level3">
<h3><span class="header-section-number">5.2.1</span> Sampling variance of the expression distance</h3>
<p>In practice, calculating the sampling variance for the estimated coefficient (<span class="math inline">\(r\)</span>) of correlation is usually carried out using the Fisher transformation, the inverse hyperbolic (artanh) of <span class="math inline">\(r\)</span>, that is, <span class="math inline">\(F\left(r\right)=0.5\ln\left(\frac{1+r}{1-r}\right)\)</span>. It follows that <span class="math inline">\(F(r)\)</span> approximately follows a normal distribution with <span class="math inline">\(F(\rho)\)</span> and the variance of <span class="math inline">\(\frac{1}{N-3}\)</span>, where <span class="math inline">\(\rho\)</span> is the true value of the coefficient of correlation and <span class="math inline">\(N\)</span> is the sample size. With the delta method, the inverse Fisher transformation brings the sampling variance back to the correlation scale, resulting in <span class="math inline">\(Var\left(r\right)=\frac{\left(1-r^2\right)^2}{N-3}\)</span>. Consider the general expression distance given by Eq.(1.11). By the delta method, the large-sampling variance of <span class="math inline">\(D_{12}\)</span> is approximately given by
<span class="math display">\[Var\left(D_{12}\right)=\frac{Var\left(r_{12}\right)}{\left(r_{12}-\pi\right)^2}\tag{2.4}\]</span></p>
</div>
<div id="calculation-of-varbigtriangleup_ab" class="section level3">
<h3><span class="header-section-number">5.2.2</span> Calculation of <span class="math inline">\(Var(\bigtriangleup_{AB})\)</span></h3>
<p>Since <span class="math inline">\(\bigtriangleup_{AB}=D_{AC}-D_{BC}\)</span>, we have <span class="math inline">\(Var(\bigtriangleup_{AB})=Var(D_{AC})+Var(D_{BC})-2Cov(D_{AC}, D_{BC})\)</span>. Moreover, we notice that the sampling covariance equals to <span class="math inline">\(Cov(D_{AC}, D_{BC})=Var(c)\)</span>, as branch-<span class="math inline">\(c\)</span> is the one shared by those two distances. We develop a simple method to calculate <span class="math inline">\(Var(c)\)</span>.</p>
<ol style="list-style-type: decimal">
<li>First, the branch length can be estimated by <span class="math inline">\(c=\frac{(D_{AC}+D_{BC}-D_{AB})}{2}\)</span>.</li>
<li>we calculate a new variable <span class="math inline">\(r_c=\pi+(1-\pi)e^{-c}\)</span>. After viewing <span class="math inline">\(r_c\)</span> as the estimated coefficient of correlation between species <span class="math inline">\(C\)</span> and the ancestral node <span class="math inline">\(O\)</span>, by the Fisher transformation we obtain <span class="math inline">\(Var(c)=\frac{(1-r_c^2)^2}{(N-3)}\)</span>. While two sampling variances <span class="math inline">\(Var(D_{AC})\)</span> and <span class="math inline">\(Var(D_{BC})\)</span> can be obtained directly from Eq.(2.4), together we have</li>
</ol>
<p><span class="math display">\[Var\left(\bigtriangleup_{AB}\right)=\frac{Var\left(r_{AC}\right)}{\left(r_{AC}-\pi\right)^2}+\frac{Var\left(r_{BC}\right)}{\left(r_{BC}-\pi\right)^2}-\frac{2Var\left(r_c\right)}{\left(r_c-\pi\right)^2}\tag{2.5}\]</span>
</p>
</div>
</div>
<div id="case-study-testing-fast-evolving-genes-in-human" class="section level2">
<h2><span class="header-section-number">5.3</span> Case Study: Testing Fast-evolving Genes in Human</h2>
<p>In general, relative rate test is to investigate whether the expression of a given gene set in species A is fastly evolved than that in species B. Typically, the gene set chosen to perform the test is set by users and is only a small part of all one-to-one orthologous genes (several hundreds). Moreover, the gene set usually have some specific biological characteristics, such like gene set from a GO term or a co-expression module.</p>
<p>In the case study, we choose gene set with row numbers between 200 to 800 to exemplify how relative rate test works.</p>
<p>At first, we shall extract the gene expression values of brain tissue from the <em>tetraExp</em> object.</p>
<p>Parameter <code>rowindex</code> defines which genes are selected to perfome the test. Usually, <code>rowindex</code> is a vector of numbers corresponded to indices of selecting rows or a vector of logical values (TRUE or FALSE) indicating whether to select the corresponding row or not.</p>
<pre class="sourceCode r"><code class="sourceCode r">#### load the tetraExp data firstly
<span class="kw">data</span>(tetraExp)
### extract the gene expression values of the selected genes from &#39;tetraExp&#39; object.
exp_table &lt;-<span class="kw">exptabTE</span>(tetraExp, <span class="dt">taxa =</span> <span class="st">&#39;all&#39;</span>, <span class="dt">subtaxa =</span> <span class="st">&#39;Brain&#39;</span>,<span class="dt">rowindex =</span> <span class="dv">200</span><span class="op">:</span><span class="dv">800</span>)</code></pre>
<p>After obtaining the expression table, performing relative rate test is straightforward by function <code>RelaRate.test</code>. It will extract the gene expression value of species <span class="math inline">\(A\)</span> (human), species <span class="math inline">\(B\)</span> (chimpanzee) and outgroup <span class="math inline">\(C\)</span> (macaque) from the <em>exp_table</em> and will return a list of three elements. The first one is the <span class="math inline">\(Z\)</span>-score value of the relative rate test, the second is the value of parameter <code>alternative</code>, and the last one is the <span class="math inline">\(p\)</span>-value under the specified alternative hypothesis.</p>
<pre class="sourceCode r"><code class="sourceCode r">ztest &lt;-<span class="st"> </span><span class="kw">RelaRate.test</span>(<span class="dt">expTable =</span> exp_table, <span class="dt">x =</span> <span class="st">&#39;human&#39;</span>, <span class="dt">y =</span> <span class="st">&#39;chimpanzee&#39;</span>,
                       <span class="dt">outgroup =</span> <span class="st">&#39;macaque&#39;</span>, <span class="dt">alternative =</span> <span class="st">&#39;greater&#39;</span>)
ztest</code></pre>
<pre><code>## $Z_score
## [1] 0.7349926
## 
## $alternative
## [1] &quot;greater&quot;
## 
## $p.value
## [1] 0.231172</code></pre>
<p>So, using macaque as an outgroup, the expression of geneset with row numbers between 200 and 800 in human is not likely to evolved significantly faster than that in chimpanzee (<span class="math inline">\(p\)</span>-value &gt;0.05).</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="pairwise-expression-distance-and-expression-tree-making.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="estimating-the-strength-of-expression-conservation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-RelativeTest.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["TreeExp-Tutorial.pdf", "TreeExp-Tutorial.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
